<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light">
<script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/3/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>Gene-level Circular Amplicon Prediction • gcap</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Gene-level Circular Amplicon Prediction">
<meta property="og:description" content="Provides data processing pipeline feeding paired bam files
    and XGBOOST model for predicting tumor circular amplicons (also known
    as ecDNA) in gene level.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/3/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/3/darkly/bootstrap.css" media="(prefers-color-scheme: dark)">
<!-- preferably CSS --><link rel="stylesheet" href="preferably.css">
<link id="css-code-light" rel="stylesheet" href="code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<link id="css-code-dark" rel="stylesheet" href="code-color-scheme-dark.css" media="(prefers-color-scheme: dark)">
<script src="darkswitch.js"></script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">gcap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
          <a href="#" id="css-toggle-btn">
            <span class="fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        <li>
  <a href="https://github.com/ShixiangWang/gcap/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
        
        


      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="gcap-gene-level-circular-amplicon-prediction">GCAP: Gene-level Circular Amplicon Prediction<a class="anchor" aria-label="anchor" href="#gcap-gene-level-circular-amplicon-prediction"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>In a nutshell, <strong>gcap</strong> provides an end-to-end workflow for predicting circular amplicon (also known as ecDNA, extra-chromosomal DNA ) in gene level with machine learning approach, then classifying cancer samples into different focal amplification (fCNA) types, based on input from WES (tumor-normal paired BAM) data, allele specific copy number data (e.g., results from <a href="https://github.com/VanLoo-lab/ascat" class="external-link">ASCAT</a> or <a href="https://cran.r-project.org/package=sequenza" class="external-link">Sequenza</a>), or even absolute integer copy number data (e.g., results from <a href="https://software.broadinstitute.org/cancer/cga/absolute" class="external-link">ABSOLUTE</a>). The former two data sources are preferred as input of <strong>gcap</strong> .</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="download-reference-files-wes-bam-data-only">Download reference files (WES bam data only)<a class="anchor" aria-label="anchor" href="#download-reference-files-wes-bam-data-only"></a>
</h3>
<p>For advanced users, you can prepare the reference files by following the instructions from <a href="https://github.com/shixiangwang/ascat/tree/v3.0" class="external-link uri">https://github.com/shixiangwang/ascat/tree/v3.0</a>.</p>
<p>We recommend all users directly download the reference files from the links below:</p>
<ul>
<li><a href="https://zenodo.org/record/5533065" class="external-link">Curated reference files for GCAP (WES)</a></li>
</ul>
<blockquote>
<p>The prediction model was built with data on the top of hg38 genome build, while hg38-based BAM file input is more recommended.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="install-allelecount-wes-bam-data-only">Install alleleCount (WES bam data only)<a class="anchor" aria-label="anchor" href="#install-allelecount-wes-bam-data-only"></a>
</h3>
<p><a href="https://github.com/cancerit/alleleCount" class="external-link"><strong>alleleCount</strong></a> is required to run <strong>ASCAT</strong> on WES bam data, if you haven’t installed <a href="https://docs.conda.io/en/latest/" class="external-link"><strong>conda</strong></a> or <a href="https://docs.conda.io/en/latest/miniconda.html" class="external-link"><strong>miniconda</strong></a>, please install firstly, then install the <strong>alleleCount</strong> in terminal with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> cancerit <span class="at">-c</span> bioconda cancerit-allelecount</span></code></pre></div>
<blockquote>
<p>NOTE: <strong>gcap</strong> set the default <strong>alleleCount</strong> as the <code>~/miniconda3/envs/cancerit/bin/alleleCounter</code>, if you use <strong>conda</strong> or other approaches, please set the path when you use corresponding functions.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="install-ascat-required">Install ASCAT (required)<a class="anchor" aria-label="anchor" href="#install-ascat-required"></a>
</h3>
<p>Install <strong>ASCAT</strong> v3.0 (modified and adapted for GCAP workflow in HPC) in R console from GitHub with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># This is a forked version ASCAT</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ShixiangWang/ascat@v3-for-gcap-v1"</span>, subdir <span class="op">=</span> <span class="st">"ASCAT"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="install-gcap-required">Install GCAP (required)<a class="anchor" aria-label="anchor" href="#install-gcap-required"></a>
</h3>
<p>Install <strong>gcap</strong> in R console from GitHub with:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ShixiangWang/gcap"</span><span class="op">)</span></code></pre></div>
<p>If you would like to use CLI program, run the following code in your R console:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gcap</span><span class="fu">::</span><span class="fu"><a href="reference/deploy.html">deploy</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Two scripts <code>gcap-bam.R</code> and <code>gcap-ascn.R</code> shall be linked to your path <code>/usr/local/bin/</code>. You can use one of them based on you input data.</p>
</div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>Run the following code to see a quick example:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ShixiangWang/gcap" class="external-link">gcap</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ascn"</span><span class="op">)</span>
<span class="va">rv</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/gcap.ASCNworkflow.html">gcap.ASCNworkflow</a></span><span class="op">(</span><span class="va">ascn</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span>
<span class="va">rv</span></code></pre></div>
<div class="section level3">
<h3 id="pipeline-wes-bam-data-only">Pipeline (WES bam data only)<a class="anchor" aria-label="anchor" href="#pipeline-wes-bam-data-only"></a>
</h3>
<ul>
<li>for one tumor-normal pair, you can refer to <a href="https://github.com/ShixiangWang/gcap/blob/master/test-workflow/one-pair.R" class="external-link">one-pair.R</a>.</li>
<li>for multiple tumor-normal pairs, you can refer to <a href="https://github.com/ShixiangWang/gcap/blob/master/test-workflow/two-pairs.R" class="external-link">two-pair.R</a>.</li>
</ul>
<p>To run <strong>gcap</strong> from bam files, a machine with <strong>at least 80GB RAM</strong> is required for the <code>allelecount</code> process. If you set multiple threads, please note the parallel computation is used in part of the workflow. You should balance the <code>nthread</code> setting and the computing power your machine provides by yourself.</p>
<p>It generally takes <code>~0.5h</code> to finish one case (tumor-normal pair).</p>
<p>In our practice, when we want to process multiple cases, set <code>nthread = 22</code> and directly let <strong>gcap</strong> handle multiple cases (instead of writing a loop yourself) is good enough.</p>
</div>
<div class="section level3">
<h3 id="pipeline-allele-specific-or-absolute-copy-number-data">Pipeline (Allele specific or absolute copy number data)<a class="anchor" aria-label="anchor" href="#pipeline-allele-specific-or-absolute-copy-number-data"></a>
</h3>
<p>Please refer to <a href="https://shixiangwang.github.io/gcap/reference/gcap.ASCNworkflow.html" class="external-link"><code>?gcap.ASCNworkflow()</code></a>.</p>
</div>
<div class="section level3">
<h3 id="functions">Functions<a class="anchor" aria-label="anchor" href="#functions"></a>
</h3>
<p>For more custom and advanced control of the analysis, you can read the structured documentation at <a href="https://shixiangwang.github.io/gcap/reference/index.html" class="external-link"><em>package site</em></a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="logging">Logging<a class="anchor" aria-label="anchor" href="#logging"></a>
</h2>
<p>For better debugging and rechecking. The logging information of your operation with <strong>gcap</strong> would be saved into an independent file. You can use the following commands to get the file path and print logging message. Please note you have to use <code>:::</code> to access these functions as they are not exported from <strong>gcap</strong>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> gcap<span class="sc">:::</span><span class="fu">get_log_file</span>()</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"~/Library/Logs/gcap/gcap.log"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> gcap<span class="sc">:::</span><span class="fu">cat_log_file</span>()</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="related-tools">Related tools<a class="anchor" aria-label="anchor" href="#related-tools"></a>
</h2>
<ul>
<li>
<a href="https://github.com/ShixiangWang/DoAbsolute" class="external-link">DoAbsolute</a>: Automate Absolute Copy Number Calling using ‘ABSOLUTE’ package.</li>
<li>
<a href="https://github.com/ShixiangWang/sigminer" class="external-link">sigminer</a>: An easy-to-use and scalable toolkit for genomic alteration signature (a.k.a. mutational signature) analysis and visualization in R.</li>
</ul>
</div>
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<p><strong>gcap</strong> outputs two data tables including feature table and prediction result.</p>
</div>
<div class="section level2">
<h2 id="citations">Citations<a class="anchor" aria-label="anchor" href="#citations"></a>
</h2>
</div>
<div class="section level2">
<h2 id="license">LICENSE<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>This software and associated documentation files (the “Software”) are protected by copyright. This Software is provided “as is” (at your own risk) for internal non-commercial academic research purposes only. Please read the <a href="LICENSE">Non-Commercial Academic License</a> in detail before downloading a copy. By installing or using this Software, you agree to be bound by the terms and conditions of the Non-Commercial Academic License.</p>
<p>All commercial use of the Software or any modification, manipulation or derivative of the Software, including but not limited to transfer, sale or licence to a commercial third party or use on behalf of a commercial third party (including but not limited to use as part of a service supplied to any third party for financial reward) is strictly prohibited and requires a commercial use licence. For further information please email <a href="mailto:wangsx1@sysucc.org.cn" class="email">wangsx1@sysucc.org.cn</a> or <a href="mailto:zhaoqi@sysucc.org.cn" class="email">zhaoqi@sysucc.org.cn</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/ShixiangWang/gcap/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ShixiangWang/gcap/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>Non-Commercial Academic License + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gcap</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-9855-7357" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
