<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light"><script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/3/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>R6 class representing focal copy number amplification list predicted from a cohort — fCNA • gcap</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="R6 class representing focal copy number amplification list predicted from a cohort — fCNA"><meta property="og:description" content="Contains fields storing data and methods to get, process and visualize
fCNA information. Examples please see gcap.ASCNworkflow()."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/3/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/3/darkly/bootstrap.css" media="(prefers-color-scheme: dark)"><!-- preferably CSS --><link rel="stylesheet" href="../preferably.css"><link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)"><script src="../darkswitch.js"></script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gcap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.11.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
          <a href="#" id="css-toggle-btn">
            <span class="fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        <li>
  <a href="https://github.com/ShixiangWang/gcap/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
        
        


      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>R6 class representing focal copy number amplification list predicted from a cohort</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/gcap/blob/HEAD/R/fCNA-object.R" class="external-link"><code>R/fCNA-object.R</code></a></small>
    <div class="hidden name"><code>fCNA.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Contains fields storing data and methods to get, process and visualize
fCNA information. Examples please see <code><a href="gcap.ASCNworkflow.html">gcap.ASCNworkflow()</a></code>.</p>
    </div>


    <div id="public-fields">
    <h2>Public fields</h2>
    <p></p><div class="r6-fields"><dl><dt><code>data</code></dt>
<dd><p>a <code>data.table</code> storing fCNA list, which typically contains following columns:</p><ul><li><p><code>sample</code> sample or case ID.</p></li>
<li><p><code>band</code> chromosome cytoband.</p></li>
<li><p><code>gene_id</code> gene ID, typically Ensembl ID. You can convert the ID with R package <code>IDConverter</code>.</p></li>
<li><p><code>total_cn</code> total copy number value.</p></li>
<li><p><code>minor_cn</code> copy number value for minor allele.</p></li>
<li><p><code>background_cn</code> background copy number value (for circular amplicon), calculated as <code>pmax(blood_cn_top5 + tightness * blood_cn_top5_sd, 2) * ploidy / 2</code>. (<strong>Optional</strong>)
At default, <code>blood_cn_top5(_sd)</code> is obtained from SNP array data of ~2000 TCGA diploidy blood samples,
they are mean and sd of top 5% copy number values for each gene (in ~2000 samples);
<code>ploidy</code> is tumor ploidy.</p></li>
<li><p><code>prob</code> the probability the gene located in circular DNA.</p></li>
<li><p><code>amplicon_type</code> the type of DNA amplicon.</p></li>
</ul></dd>


<dt><code>sample_summary</code></dt>
<dd><p>a <code>data.table</code> storing sample summary data, which typically contains
at least the following columns:</p><ul><li><p><code>sample</code> sample or case ID. <strong>Should only include cases have been called with GCAP workflow,
otherwise the extra cases would be automatically classified as 'nofocal' (i.e. <code>NA</code> in <code>sample_summary</code> field) class</strong>.</p></li>
<li><p><code>purity</code>, <code>ploidy</code> for tumor purity or ploidy.</p></li>
<li><p><code>AScore</code> aneuploidy score.</p></li>
<li><p><code>pLOH</code> genome percentage harboring LOH events.</p></li>
<li><p><code>CN1 ... CN19</code> activity of copy number signatures.</p></li>
<li><p><strong><code>class</code></strong> the sample class based on amplicon type.</p></li>
<li><p><code>ec_genes</code> number of genes predicted as located on circular DNA.</p></li>
<li><p><code>ec_possibly_genes</code> same with <code>ec_genes</code> but with less confidence.</p></li>
<li><p><code>ec_cytobands</code> number of cytobands predicted as located on circular DNA.
(the regions of <code>ec_possibly_genes</code> are not included in computation)</p></li>
</ul></dd>


<dt><code>gene_summary</code></dt>
<dd><p>a <code>data.table</code> storing gene summary data.</p></dd>


<dt><code>cytoband_summary</code></dt>
<dd><p>a <code>data.table</code> storing cytoband summary data.</p></dd>


</dl><p></p></div>
    </div>
    <div id="active-bindings">
    <h2>Active bindings</h2>
    <p></p><div class="r6-active-bindings"><dl><dt><code>min_n</code></dt>
<dd><p>check <code>$new()</code> method for details. If you updated this value,
a function will be called to update the sample summary.</p></dd>


</dl><p></p></div>
    </div>
    <div id="methods">
    <h2>Methods</h2>
    
<div class="section">
<h3 id="public-methods">Public methods<a class="anchor" aria-label="anchor" href="#public-methods"></a></h3>

<ul><li><p><a href="#method-new"><code>fCNA$new()</code></a></p></li>
<li><p><a href="#method-subset"><code>fCNA$subset()</code></a></p></li>
<li><p><a href="#method-getSampleSummary"><code>fCNA$getSampleSummary()</code></a></p></li>
<li><p><a href="#method-getGeneSummary"><code>fCNA$getGeneSummary()</code></a></p></li>
<li><p><a href="#method-getCytobandSummary"><code>fCNA$getCytobandSummary()</code></a></p></li>
<li><p><a href="#method-saveToFiles"><code>fCNA$saveToFiles()</code></a></p></li>
<li><p><a href="#method-convertGeneID"><code>fCNA$convertGeneID()</code></a></p></li>
<li><p><a href="#method-print"><code>fCNA$print()</code></a></p></li>
</ul></div><p></p><hr><a id="method-new"></a><div class="section">
<h3 id="method-new-">Method <code>new()</code><a class="anchor" aria-label="anchor" href="#method-new-"></a></h3>
<p>Create a <code>fCNA</code> object.
Typically, you can obtain this object from <code><a href="gcap.workflow.html">gcap.workflow()</a></code> or <code><a href="gcap.ASCNworkflow.html">gcap.ASCNworkflow()</a></code>.</p><div class="section">
<h4 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va"><a href="../reference/fCNA.html">fCNA</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="va">fcna</span>, pdata <span class="op">=</span> <span class="va">fcna</span><span class="op">[</span>, <span class="st">"sample"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, min_n <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h4>
<p></p><div class="arguments"><dl><dt><code>fcna</code></dt>
<dd><p>a <code>data.frame</code> storing focal copy number amplicon list.</p></dd>


<dt><code>pdata</code></dt>
<dd><p>a <code>data.frame</code> storing phenotype or sample-level related data. (Optional)</p></dd>


<dt><code>min_n</code></dt>
<dd><p>a minimal cytoband number (default is <code>1</code>) to determine
sample class. e.g., sample with at least 1 cytoband harboring circular
genes would be labelled as "circular". <strong>This only affect <code>sample_summary</code> field</strong>.</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-subset"></a><div class="section">
<h3 id="method-subset-">Method <code><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset()</a></code><a class="anchor" aria-label="anchor" href="#method-subset-"></a></h3>
<p>Return a subset <code>fCNA</code> object</p><div class="section">
<h4 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va">fCNA</span><span class="op">$</span><span class="fu">subset</span><span class="op">(</span><span class="va">...</span>, on <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"sample_summary"</span><span class="op">)</span><span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-1">Arguments<a class="anchor" aria-label="anchor" href="#arguments-1"></a></h4>
<p></p><div class="arguments"><dl><dt><code>...</code></dt>
<dd><p>subset expressions on <code>fCNA$data</code> or <code>fCNA$sample_summary</code>.</p></dd>


<dt><code>on</code></dt>
<dd><p>if it is "data", subset operations are on data field of <code>fCNA</code> object,
same for "sample_summary".</p></dd>


</dl><p></p></div>
</div>
<div class="section">
<h4 id="returns">Returns<a class="anchor" aria-label="anchor" href="#returns"></a></h4>
<p>a <code>fCNA</code></p>
</div>

</div><p></p><hr><a id="method-getSampleSummary"></a><div class="section">
<h3 id="method-getsamplesummary-">Method <code>getSampleSummary()</code><a class="anchor" aria-label="anchor" href="#method-getsamplesummary-"></a></h3>
<p>Get sample summary of fCNA</p><div class="section">
<h4 id="usage-2">Usage<a class="anchor" aria-label="anchor" href="#usage-2"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va">fCNA</span><span class="op">$</span><span class="fu">getSampleSummary</span><span class="op">(</span><span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="returns-1">Returns<a class="anchor" aria-label="anchor" href="#returns-1"></a></h4>
<p>a <code>data.table</code></p>
</div>

</div><p></p><hr><a id="method-getGeneSummary"></a><div class="section">
<h3 id="method-getgenesummary-">Method <code>getGeneSummary()</code><a class="anchor" aria-label="anchor" href="#method-getgenesummary-"></a></h3>
<p>Get gene level summary of fCNA type</p><div class="section">
<h4 id="usage-3">Usage<a class="anchor" aria-label="anchor" href="#usage-3"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va">fCNA</span><span class="op">$</span><span class="fu">getGeneSummary</span><span class="op">(</span><span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="returns-2">Returns<a class="anchor" aria-label="anchor" href="#returns-2"></a></h4>
<p>a <code>data.table</code></p>
</div>

</div><p></p><hr><a id="method-getCytobandSummary"></a><div class="section">
<h3 id="method-getcytobandsummary-">Method <code>getCytobandSummary()</code><a class="anchor" aria-label="anchor" href="#method-getcytobandsummary-"></a></h3>
<p>Get cytoband level summary of fCNA type</p><div class="section">
<h4 id="usage-4">Usage<a class="anchor" aria-label="anchor" href="#usage-4"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va">fCNA</span><span class="op">$</span><span class="fu">getCytobandSummary</span><span class="op">(</span><span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="returns-3">Returns<a class="anchor" aria-label="anchor" href="#returns-3"></a></h4>
<p>a <code>data.table</code></p>
</div>

</div><p></p><hr><a id="method-saveToFiles"></a><div class="section">
<h3 id="method-savetofiles-">Method <code>saveToFiles()</code><a class="anchor" aria-label="anchor" href="#method-savetofiles-"></a></h3>
<p>Save the key data to local files</p><div class="section">
<h4 id="usage-5">Usage<a class="anchor" aria-label="anchor" href="#usage-5"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va">fCNA</span><span class="op">$</span><span class="fu">saveToFiles</span><span class="op">(</span><span class="va">dirpath</span>, fileprefix <span class="op">=</span> <span class="st">"fCNA"</span><span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-2">Arguments<a class="anchor" aria-label="anchor" href="#arguments-2"></a></h4>
<p></p><div class="arguments"><dl><dt><code>dirpath</code></dt>
<dd><p>directory path storing output files.</p></dd>


<dt><code>fileprefix</code></dt>
<dd><p>file prefix. Two result files shall be generated.</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-convertGeneID"></a><div class="section">
<h3 id="method-convertgeneid-">Method <code>convertGeneID()</code><a class="anchor" aria-label="anchor" href="#method-convertgeneid-"></a></h3>
<p>Convert Gene IDs between Ensembl and Hugo Symbol System</p><div class="section">
<h4 id="usage-6">Usage<a class="anchor" aria-label="anchor" href="#usage-6"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va">fCNA</span><span class="op">$</span><span class="fu">convertGeneID</span><span class="op">(</span>
  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl"</span>, <span class="st">"symbol"</span><span class="op">)</span>,
  genome_build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hg38"</span>, <span class="st">"hg19"</span>, <span class="st">"mm10"</span>, <span class="st">"mm9"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-3">Arguments<a class="anchor" aria-label="anchor" href="#arguments-3"></a></h4>
<p></p><div class="arguments"><dl><dt><code>type</code></dt>
<dd><p>type of input IDs, could be 'ensembl' or 'symbol'.</p></dd>


<dt><code>genome_build</code></dt>
<dd><p>reference genome build.</p></dd>


</dl><p></p></div>
</div>

</div><p></p><hr><a id="method-print"></a><div class="section">
<h3 id="method-print-">Method <code><a href="https://rdrr.io/r/base/print.html" class="external-link">print()</a></code><a class="anchor" aria-label="anchor" href="#method-print-"></a></h3>
<p>print the fCNA object</p><div class="section">
<h4 id="usage-7">Usage<a class="anchor" aria-label="anchor" href="#usage-7"></a></h4>
<p></p><div class="r"><div class="sourceCode"><pre><code><span class="va">fCNA</span><span class="op">$</span><span class="fu">print</span><span class="op">(</span><span class="va">...</span><span class="op">)</span></code></pre></div><p></p></div>
</div>

<div class="section">
<h4 id="arguments-4">Arguments<a class="anchor" aria-label="anchor" href="#arguments-4"></a></h4>
<p></p><div class="arguments"><dl><dt><code>...</code></dt>
<dd><p>unused.</p></dd>


</dl><p></p></div>
</div>

</div>

    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

      </footer></div>

  


  

  </body></html>

