<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light"><script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/3/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>Draw Oncoprint for fCNA Profile Visualization — gcap.plotProfile • gcap</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Draw Oncoprint for fCNA Profile Visualization — gcap.plotProfile"><meta property="og:description" content="See oncoprint
and heatmap
to modify the result plot in depth."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/3/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/3/darkly/bootstrap.css" media="(prefers-color-scheme: dark)"><!-- preferably CSS --><link rel="stylesheet" href="../preferably.css"><link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)"><script src="../darkswitch.js"></script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gcap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.10.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
          <a href="#" id="css-toggle-btn">
            <span class="fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        <li>
  <a href="https://github.com/ShixiangWang/gcap/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
        
        


      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Draw Oncoprint for fCNA Profile Visualization</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/gcap/blob/HEAD/R/plotProfile.R" class="external-link"><code>R/plotProfile.R</code></a></small>
    <div class="hidden name"><code>gcap.plotProfile.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>See <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/oncoprint.html" class="external-link">oncoprint</a>
and <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/a-single-heatmap.html#size-of-the-heatmap" class="external-link">heatmap</a>
to modify the result plot in depth.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">gcap.plotProfile</span><span class="op">(</span>
  <span class="va">fCNA</span>,
  genes <span class="op">=</span> <span class="cn">NULL</span>,
  samples <span class="op">=</span> <span class="cn">NULL</span>,
  merge_circular <span class="op">=</span> <span class="cn">FALSE</span>,
  show_column_names <span class="op">=</span> <span class="cn">TRUE</span>,
  remove_empty_columns <span class="op">=</span> <span class="cn">FALSE</span>,
  remove_empty_rows <span class="op">=</span> <span class="cn">TRUE</span>,
  heatmap_legend_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"fCNA"</span><span class="op">)</span>,
  col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>noncircular <span class="op">=</span> <span class="st">"blue"</span>, circular <span class="op">=</span> <span class="st">"red"</span>, possibly_circular <span class="op">=</span> <span class="st">"magenta"</span><span class="op">)</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>fCNA</dt>
<dd><p>a <a href="fCNA.html">fCNA</a> object.</p></dd>
<dt>genes</dt>
<dd><p>a list of genes. Must be subsets of <code>fCNA$data$gene_id</code>.</p></dd>
<dt>samples</dt>
<dd><p>a vector to subset samples.</p></dd>
<dt>merge_circular</dt>
<dd><p>if <code>TRUE</code>, merge 'possibly_circular' into 'circular' type.</p></dd>
<dt>show_column_names</dt>
<dd><p>see <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/oncoPrint.html" class="external-link">?ComplexHeatmap::oncoPrint</a></code>.</p></dd>
<dt>remove_empty_columns</dt>
<dd><p>see <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/oncoPrint.html" class="external-link">?ComplexHeatmap::oncoPrint</a></code>.</p></dd>
<dt>remove_empty_rows</dt>
<dd><p>see <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/oncoPrint.html" class="external-link">?ComplexHeatmap::oncoPrint</a></code>.</p></dd>
<dt>heatmap_legend_param</dt>
<dd><p>see <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/oncoPrint.html" class="external-link">?ComplexHeatmap::oncoPrint</a></code>.</p></dd>
<dt>col</dt>
<dd><p>see <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/oncoPrint.html" class="external-link">?ComplexHeatmap::oncoPrint</a></code>.</p></dd>
<dt>...</dt>
<dd><p>Other parameters passing to <code><a href="https://rdrr.io/pkg/ComplexHeatmap/man/oncoPrint.html" class="external-link">ComplexHeatmap::oncoPrint</a></code>.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a <code>Heatmap</code> object, you can go further modify it.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="fCNA.html">fCNA</a> for building object.</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># \donttest{</span></span>
<span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">"ComplexHeatmap"</a></span><span class="op">)</span> <span class="op">&amp;&amp;</span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/ShixiangWang/IDConverter" class="external-link">"IDConverter"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">  <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ascn"</span><span class="op">)</span></span>
<span class="r-in">  <span class="va">data</span> <span class="op">&lt;-</span> <span class="va">ascn</span></span>
<span class="r-in"></span>
<span class="r-in">  <span class="co"># Create fake data</span></span>
<span class="r-in">  <span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span class="r-in">  <span class="va">data</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in">  <span class="va">rv</span> <span class="op">&lt;-</span> <span class="fu"><a href="gcap.ASCNworkflow.html">gcap.ASCNworkflow</a></span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">  <span class="fu">gcap.plotProfile</span><span class="op">(</span><span class="va">rv</span>, samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B"</span>, <span class="st">"A"</span>, <span class="st">"C"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">  <span class="va">rv</span><span class="op">$</span><span class="fu">convertGeneID</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in">  <span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu">gcap.plotProfile</span><span class="op">(</span><span class="va">rv</span>,</span>
<span class="r-in">    samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B"</span>, <span class="st">"A"</span>, <span class="st">"C"</span>, <span class="st">"D"</span>, <span class="st">"F"</span><span class="op">)</span>,</span>
<span class="r-in">    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,</span>
<span class="r-in">    top_annotation <span class="op">=</span> <span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html" class="external-link">HeatmapAnnotation</a></span><span class="op">(</span></span>
<span class="r-in">      cbar <span class="op">=</span> <span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_oncoprint_barplot.html" class="external-link">anno_oncoprint_barplot</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span class="r-in">      foo1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"g1"</span>, <span class="st">"g1"</span>, <span class="st">"g2"</span>, <span class="st">"g2"</span>, <span class="st">"g3"</span><span class="op">)</span>,</span>
<span class="r-in">      bar1 <span class="op">=</span> <span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_points.html" class="external-link">anno_points</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in">  <span class="op">)</span></span>
<span class="r-in">  <span class="va">ht</span></span>
<span class="r-in">  <span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">ht</span>, merge_legends <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">  <span class="co">#----------------</span></span>
<span class="r-in">  <span class="co"># Plot KM curve</span></span>
<span class="r-in">  <span class="co">#----------------</span></span>
<span class="r-in">  <span class="va">surv_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span class="r-in">    sample <span class="op">=</span> <span class="va">rv</span><span class="op">$</span><span class="va">sample_summary</span><span class="op">$</span><span class="va">sample</span>,</span>
<span class="r-in">    time <span class="op">=</span> <span class="fl">3000</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">sample_summary</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span class="r-in">    status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">sample_summary</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in">  <span class="op">)</span></span>
<span class="r-in">  <span class="va">surv_data</span></span>
<span class="r-in">  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="gcap.plotKMcurve.html">gcap.plotKMcurve</a></span><span class="op">(</span><span class="va">rv</span>, <span class="va">surv_data</span><span class="op">)</span></span>
<span class="r-in">  <span class="va">p</span></span>
<span class="r-in"></span>
<span class="r-in">  <span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="gcap.plotKMcurve.html">gcap.plotKMcurve</a></span><span class="op">(</span><span class="va">rv</span>, <span class="va">surv_data</span>, genes <span class="op">=</span> <span class="st">"MYC"</span><span class="op">)</span></span>
<span class="r-in">  <span class="va">p2</span></span>
<span class="r-in"></span>
<span class="r-in">  <span class="co"># ---------------</span></span>
<span class="r-in">  <span class="co"># Plot forest</span></span>
<span class="r-in">  <span class="co"># ---------------</span></span>
<span class="r-in">  <span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="gcap.plotForest.html">gcap.plotForest</a></span><span class="op">(</span><span class="va">rv</span>, <span class="va">surv_data</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">  <span class="co"># Fake some data for gene analysis</span></span>
<span class="r-in">  <span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">[</span><span class="va">amplicon_type</span> <span class="op">==</span> <span class="st">"circular"</span><span class="op">]</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"MYC"</span></span>
<span class="r-in">  <span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">[</span><span class="va">amplicon_type</span> <span class="op">==</span> <span class="st">"circular"</span><span class="op">]</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fl">11</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"EGFR"</span></span>
<span class="r-in">  <span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">[</span><span class="va">amplicon_type</span> <span class="op">==</span> <span class="st">"circular"</span><span class="op">]</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fl">31</span><span class="op">:</span><span class="fl">35</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"GSX2"</span></span>
<span class="r-in">  <span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="gcap.plotForest.html">gcap.plotForest</a></span><span class="op">(</span><span class="va">rv</span>, <span class="va">surv_data</span>,</span>
<span class="r-in">    x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TP53"</span>, <span class="st">"MYC"</span>, <span class="st">"ABC"</span>, <span class="st">"EGFR"</span>, <span class="st">"GSX2"</span><span class="op">)</span>, x_is_gene <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span class="r-in">    ref_line <span class="op">=</span> <span class="fl">1</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span><span class="op">)</span></span>
<span class="r-in">  <span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: ComplexHeatmap</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: grid</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ComplexHeatmap version 2.8.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Github page: https://github.com/jokergoo/ComplexHeatmap</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> If you use it in published research, please cite:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   genomic data. Bioinformatics 2016.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The new InteractiveComplexHeatmap package can directly export static </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This message can be suppressed by:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   suppressPackageStartupMessages(library(ComplexHeatmap))</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IDConverter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runASCATBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">0.115 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">When you use method 'S', please make sure you have set 'join_adj_seg' to FALSE and 'add_loh' to TRUE in 'read_copynumber() in the previous step!</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Matrix generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">0.02 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Signature index detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature database in package.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature index.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Valid index for db 'CNS_TCGA':</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">CN1 CN2 CN3 CN4 CN5 CN6 CN7 CN8 CN9 CN10 CN11 CN12 CN13 CN14 CN15 CN16 CN17 CN18 CN19</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Database and index checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Signature normalized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking row number for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking rownames for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Method 'QP' detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Corresponding function generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Calling function.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: A</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: B</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: C</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: D</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: E</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: F</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: G</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: H</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: I</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: J</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">Generating output signature exposures.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> <span style="font-weight: bold;">[2022-04-06 19:47:48]: </span><span style="color: #0000BB; font-weight: bold;">0.05 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runASCATBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runASCATBuildflow]: None exist, set to NULL.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runASCATBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [collapse_to_genes]: reading reference file D:/Rlib/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runPrediction]: using model file XGB_NF11.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19:47:48] WARNING: amalgamation/../src/c_api/c_api.cc:718: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:48 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> set fixed levels for 'amplicon_type'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing gene and cytoband...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">10</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     gene: <span style="color: #00BB00;">155</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     fCNA: <span style="color: #00BB00;">155</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ <span style="color: #00BB00;">104</span> (<span style="color: #00BBBB;">3</span>) noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ <span style="color: #00BB00;">0</span> (<span style="color: #00BBBB;">0</span>) possibly_circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ <span style="color: #00BB00;">51</span> (<span style="color: #00BBBB;">4</span>) circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:51 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:51 info [gcap.ASCNworkflow]: Saving raw prediction result to C:\Users\ADMINI~1\AppData\Local\Temp\Rtmp61Kg9u/gcap_63b28486-b59f-11ec-8000-cb3394b9d359_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:52 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to C:\Users\ADMINI~1\AppData\Local\Temp\Rtmp61Kg9u/gcap_63b28486-b59f-11ec-8000-cb3394b9d359_fCNA_records.csv, C:\Users\ADMINI~1\AppData\Local\Temp\Rtmp61Kg9u/gcap_63b28486-b59f-11ec-8000-cb3394b9d359_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:52 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:52 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-04-06 19:47:52 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in `[.data.frame`(data, , samples, drop = FALSE):</span> undefined columns selected</span>
<span class="r-in"><span class="co"># }</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

      </footer></div>

  


  

  </body></html>

