<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light"><script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/3/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>GCAP workflow for gene-level amplicon prediction from ASCN input — gcap.ASCNworkflow • gcap</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="GCAP workflow for gene-level amplicon prediction from ASCN input — gcap.ASCNworkflow"><meta property="og:description" content="Unlike gcap.workflow, this function directly uses the allele-specific
copy number data along with some extra sample information to infer
ecDNA genes."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/3/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/3/darkly/bootstrap.css" media="(prefers-color-scheme: dark)"><!-- preferably CSS --><link rel="stylesheet" href="../preferably.css"><link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)"><script src="../darkswitch.js"></script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gcap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
          <a href="#" id="css-toggle-btn">
            <span class="fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        <li>
  <a href="https://github.com/ShixiangWang/gcap/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
        
        


      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GCAP workflow for gene-level amplicon prediction from ASCN input</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/gcap/blob/HEAD/R/ASCNworkflow.R" class="external-link"><code>R/ASCNworkflow.R</code></a></small>
    <div class="hidden name"><code>gcap.ASCNworkflow.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Unlike <a href="gcap.workflow.html">gcap.workflow</a>, this function directly uses the allele-specific
copy number data along with some extra sample information to infer
ecDNA genes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">gcap.ASCNworkflow</span><span class="op">(</span>
  <span class="va">data</span>,
  genome_build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hg38"</span>, <span class="st">"hg19"</span><span class="op">)</span>,
  model <span class="op">=</span> <span class="st">"XGB11"</span>,
  tightness <span class="op">=</span> <span class="fl">1L</span>,
  gap_cn <span class="op">=</span> <span class="fl">3L</span>,
  overlap <span class="op">=</span> <span class="fl">1</span>,
  only_oncogenes <span class="op">=</span> <span class="cn">FALSE</span>,
  outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,
  result_file_prefix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gcap_"</span>, <span class="fu">uuid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/uuid/man/UUIDgenerate.html" class="external-link">UUIDgenerate</a></span><span class="op">(</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data</dt>
<dd><p>a <code>data.frame</code> with following columns. The key columns can be obtained
from common allele specific CNV calling software, e.g., ASCAT, Sequenza, FACETS.</p><ul><li><p>chromosome: chromosome names starts with 'chr'.</p></li>
<li><p>start: start position of the segment.</p></li>
<li><p>end: end position of the segment.</p></li>
<li><p>total_cn: total integer copy number of the segment.</p></li>
<li><p>minor_cn: minor allele integer copy number of the segment. Set it
to <code>NA</code> if you don't have this data.</p></li>
<li><p>sample: sample identifier.</p></li>
<li><p>purity: tumor purity of the sample. Set to <code>1</code> if you don't know.</p></li>
<li><p>ploidy (optinal): ploidy value of the sample tumor genome.</p></li>
<li><p>age (optional): age of the case, use along with <code>gender</code>.</p></li>
<li><p>gender (optional): gender of the case, use along with <code>age</code>.</p></li>
<li><p>type (optional): cancer type of the case, use along with <code>age</code> and <code>gender</code>.
Please refer to <a href="gcap.collapse2Genes.html">gcap.collapse2Genes</a> to see the supported cancer types.
This info is only used in 'XGB56' model. If you don't use this model, you
don't need to set it.</p></li>
</ul></dd>
<dt>genome_build</dt>
<dd><p>genome build version, should be
one of 'hg38', 'hg19'.</p></dd>
<dt>model</dt>
<dd><p>model name ("XGB11", "XGB32", "XGB56") or a custom model
from input. 'toy' can be used for test.</p></dd>
<dt>tightness</dt>
<dd><p>a coefficient to times to TCGA somatic CN to set a more strict threshold
as a circular amplicon.
If the value is larger, it is more likely a fCNA assigned to <code>noncircular</code>
instead of <code>circular</code>. <strong>When it is <code>NA</code>, we don't use TCGA somatic CN data as reference</strong>.</p></dd>
<dt>gap_cn</dt>
<dd><p>a gap copy number value, default is <code>4L</code> refer to Kim 2020 Nat.Gen.
A gene with copy number above <code>ploidy + gap_cn</code> would be treated as focal amplicon.
Smaller, more amplicons.</p></dd>
<dt>overlap</dt>
<dd><p>the overlap percentage on gene.</p></dd>
<dt>only_oncogenes</dt>
<dd><p>if <code>TRUE</code>, only known oncogenes are kept for fCNA analysis
and circular prediction.</p></dd>
<dt>outdir</dt>
<dd><p>result output path.</p></dd>
<dt>result_file_prefix</dt>
<dd><p>file name prefix (without directory path) for storing
final model prediction file in CSV format.
Default a unique file name is generated by UUID approach.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a list of invisible <code>data.table</code> and corresponding files saved to local machine.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ascn"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">ascn</span></span>
<span class="r-in"><span class="va">rv</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.runASCATBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:39 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:39]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">0.142 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">When you use method 'S', please make sure you have set 'join_adj_seg' to FALSE and 'add_loh' to TRUE in 'read_copynumber() in the previous step!</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Matrix generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">0.025 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Signature index detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature database in package.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature index.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Valid index for db 'CNS_TCGA':</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">CN1 CN2 CN3 CN4 CN5 CN6 CN7 CN8 CN9 CN10 CN11 CN12 CN13 CN14 CN15 CN16 CN17 CN18 CN19</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Database and index checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Signature normalized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Checking row number for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Checking rownames for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Method 'QP' detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Corresponding function generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Calling function.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-02-2485-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-05-4422-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">Generating output signature exposures.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:40]: </span><span style="color: #0000BB; font-weight: bold;">0.266 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runASCATBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runASCATBuildflow]: None exist, set to NULL.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runASCATBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [collapse_to_genes]: reading reference file D:/Rlib/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runPrediction]: using model file XGB_NF11.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11:34:40] WARNING: amalgamation/../src/c_api/c_api.cc:718: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:40 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runScoring]: only keep genes with copy number &gt;= ploidy+2 in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">255</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) possibly_circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: Saving raw prediction result to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_00abd8d8-37cc-11ed-8000-03c38a3f6043_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_00abd8d8-37cc-11ed-8000-03c38a3f6043_fCNA_records.csv, C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_00abd8d8-37cc-11ed-8000-03c38a3f6043_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span class="va">data</span><span class="op">$</span><span class="va">purity</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span class="r-in"><span class="va">rv2</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runASCATBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">0.092 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">When you use method 'S', please make sure you have set 'join_adj_seg' to FALSE and 'add_loh' to TRUE in 'read_copynumber() in the previous step!</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Matrix generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">0.021 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Signature index detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature database in package.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature index.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Valid index for db 'CNS_TCGA':</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">CN1 CN2 CN3 CN4 CN5 CN6 CN7 CN8 CN9 CN10 CN11 CN12 CN13 CN14 CN15 CN16 CN17 CN18 CN19</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Database and index checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Signature normalized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Checking row number for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Checking rownames for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Method 'QP' detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Corresponding function generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Calling function.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-02-2485-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-05-4422-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">Generating output signature exposures.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:44]: </span><span style="color: #0000BB; font-weight: bold;">0.044 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runASCATBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runASCATBuildflow]: None exist, set to NULL.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runASCATBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [collapse_to_genes]: reading reference file D:/Rlib/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runPrediction]: using model file XGB_NF11.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11:34:44] WARNING: amalgamation/../src/c_api/c_api.cc:718: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:44 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:45 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:45 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:45 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:45 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:47 info [gcap.runScoring]: only keep genes with copy number &gt;= ploidy+2 in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">255</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) possibly_circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:47 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:47 info [gcap.ASCNworkflow]: Saving raw prediction result to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_02dae75c-37cc-11ed-8000-03c38a3f6043_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_02dae75c-37cc-11ed-8000-03c38a3f6043_fCNA_records.csv, C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_02dae75c-37cc-11ed-8000-03c38a3f6043_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span class="va">data</span><span class="op">$</span><span class="va">age</span> <span class="op">&lt;-</span> <span class="fl">60</span></span>
<span class="r-in"><span class="va">data</span><span class="op">$</span><span class="va">gender</span> <span class="op">&lt;-</span> <span class="st">"XY"</span></span>
<span class="r-in"><span class="va">rv3</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB32"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runASCATBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">0.094 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">When you use method 'S', please make sure you have set 'join_adj_seg' to FALSE and 'add_loh' to TRUE in 'read_copynumber() in the previous step!</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Matrix generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">0.021 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Signature index detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature database in package.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature index.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Valid index for db 'CNS_TCGA':</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">CN1 CN2 CN3 CN4 CN5 CN6 CN7 CN8 CN9 CN10 CN11 CN12 CN13 CN14 CN15 CN16 CN17 CN18 CN19</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Database and index checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Signature normalized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking row number for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Checking rownames for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Method 'QP' detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Corresponding function generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Calling function.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-02-2485-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-05-4422-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">Generating output signature exposures.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:48]: </span><span style="color: #0000BB; font-weight: bold;">0.044 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runASCATBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runASCATBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: checking data type for age column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: checking data type for gender column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 warn [gcap.collapse2Genes]: non numeric type found, transforming XY to 1, otherwise 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [collapse_to_genes]: reading reference file D:/Rlib/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: merging data and extra info</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runPrediction]: using model file XGB_NF32.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11:34:48] WARNING: amalgamation/../src/c_api/c_api.cc:718: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:48 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.runScoring]: only keep genes with copy number &gt;= ploidy+2 in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">255</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) possibly_circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: Saving raw prediction result to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_04e8c6fe-37cc-11ed-8000-03c38a3f6043_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_04e8c6fe-37cc-11ed-8000-03c38a3f6043_fCNA_records.csv, C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_04e8c6fe-37cc-11ed-8000-03c38a3f6043_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span class="co"># If you want to use 'XGB56', you should include 'type' column</span></span>
<span class="r-in"><span class="va">data</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">"LUAD"</span></span>
<span class="r-in"><span class="va">rv4</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB56"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.runASCATBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">0.095 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">When you use method 'S', please make sure you have set 'join_adj_seg' to FALSE and 'add_loh' to TRUE in 'read_copynumber() in the previous step!</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Matrix generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">0.022 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Signature index detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature database in package.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Checking signature index.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Valid index for db 'CNS_TCGA':</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">CN1 CN2 CN3 CN4 CN5 CN6 CN7 CN8 CN9 CN10 CN11 CN12 CN13 CN14 CN15 CN16 CN17 CN18 CN19</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Database and index checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Signature normalized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Checking row number for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Checking rownames for catalog matrix and signature matrix.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Method 'QP' detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Corresponding function generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Calling function.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-02-2485-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Fitting sample: TCGA-05-4422-01</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">Generating output signature exposures.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:51]: </span><span style="color: #0000BB; font-weight: bold;">0.044 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.runASCATBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.runASCATBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.collapse2Genes]: one-hot encoding cancer type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:51 info [gcap.collapse2Genes]: valid cancer types: BLCA,BRCA,CESC,COAD,DLBC,ESCA,GBM,HNSC,KICH,KIRC,KIRP,LGG,LIHC,LUAD,LUSC,OV,PRAD,READ,SARC,SKCM,STAD,THCA,UCEC,UVM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.collapse2Genes]: checking data type for age column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.collapse2Genes]: checking data type for gender column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 warn [gcap.collapse2Genes]: non numeric type found, transforming XY to 1, otherwise 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [collapse_to_genes]: reading reference file D:/Rlib/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.collapse2Genes]: merging data and extra info</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.runPrediction]: using model file XGB_NF56.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11:34:52] WARNING: amalgamation/../src/c_api/c_api.cc:718: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:52 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.runScoring]: only keep genes with copy number &gt;= ploidy+2 in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">255</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) possibly_circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: Saving raw prediction result to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_0745c302-37cc-11ed-8000-03c38a3f6043_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_0745c302-37cc-11ed-8000-03c38a3f6043_fCNA_records.csv, C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_0745c302-37cc-11ed-8000-03c38a3f6043_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span class="co"># If you only have total integer copy number</span></span>
<span class="r-in"><span class="va">data</span><span class="op">$</span><span class="va">minor_cn</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span class="r-in"><span class="va">rv5</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.runASCATBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2022-09-19 11:34:55]: </span><span style="color: #0000BB; font-weight: bold;">0.085 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: pLOH score is skipped as no LOH data available</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: copy number signature activity quantification is skipped as no LOH data available</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.runASCATBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.runASCATBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: one-hot encoding cancer type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: valid cancer types: BLCA,BRCA,CESC,COAD,DLBC,ESCA,GBM,HNSC,KICH,KIRC,KIRP,LGG,LIHC,LUAD,LUSC,OV,PRAD,READ,SARC,SKCM,STAD,THCA,UCEC,UVM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: checking data type for age column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: checking data type for gender column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 warn [gcap.collapse2Genes]: non numeric type found, transforming XY to 1, otherwise 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [collapse_to_genes]: reading reference file D:/Rlib/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: merging data and extra info</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.runPrediction]: using model file XGB_NF11.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:55 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11:34:56] WARNING: amalgamation/../src/c_api/c_api.cc:718: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:56 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:58 info [gcap.runScoring]: only keep genes with copy number &gt;= ploidy+2 in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">255</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) possibly_circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:58 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:58 info [gcap.ASCNworkflow]: Saving raw prediction result to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_0957cb40-37cc-11ed-8000-03c38a3f6043_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:59 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_0957cb40-37cc-11ed-8000-03c38a3f6043_fCNA_records.csv, C:\Users\ADMINI~1\AppData\Local\Temp\RtmpKEtn9x/gcap_0957cb40-37cc-11ed-8000-03c38a3f6043_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:59 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:59 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2022-09-19 11:34:59 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># R6 class fCNA --------------------------------</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">255</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) possibly_circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               sample        band         gene_id total_cn minor_cn ploidy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1: TCGA-02-2485-01 chr12:q13.3 ENSG00000111087       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   2: TCGA-02-2485-01 chr12:q13.3 ENSG00000123329       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3: TCGA-02-2485-01 chr12:q13.3 ENSG00000135454       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4: TCGA-02-2485-01 chr12:q13.3 ENSG00000139269       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5: TCGA-02-2485-01 chr12:q13.3 ENSG00000155980       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ---                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 251: TCGA-05-4422-01  chr8:q24.3 ENSG00000204882        5        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 252: TCGA-05-4422-01  chr8:q24.3 ENSG00000226490        5        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 253: TCGA-05-4422-01  chr8:q24.3 ENSG00000234616        5        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 254: TCGA-05-4422-01  chr8:q24.3 ENSG00000283992        5        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 255: TCGA-05-4422-01  chr8:q24.3 ENSG00000284505        5        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           prob thlevel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1: 0.9875240      l3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   2: 0.9875240      l3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3: 0.9875240      l3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4: 0.9875240      l3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5: 0.9875240      l3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ---                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 251: 0.2838656      l0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 252: 0.2838656      l0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 253: 0.2838656      l0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 254: 0.2838656      l0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 255: 0.2838656      l0</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">sample_summary</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sample purity ploidy AScore       pLOH cna_burden      CN1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: TCGA-02-2485-01      1   4.09     13 0.04766306      0.951  0.00000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2: TCGA-05-4422-01      1   2.09     13 0.13041086      0.249 23.84798</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          CN2 CN3 CN4 CN5 CN6      CN7 CN8    CN9 CN10     CN11 CN12 CN13 CN14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: 45.112591   0   0   0   0 9.887409   0  0.000    0 0.000000    0    0    0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:  2.326143   0   0   0   0 8.939099   0 29.688    0 1.154688    0    0    0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        CN15 CN16 CN17     CN18 CN19    class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: 0.000000    0    0 0.000000    0 circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2: 1.178945    0    0 6.865139    0 circular</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">gene_summary</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">cytoband_summary</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Create a subset fCNA</span></span>
<span class="r-in"><span class="va">rv_subset</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">subset</span><span class="op">(</span><span class="va">total_cn</span> <span class="op">&gt;</span> <span class="fl">10</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   overlapping columns detected, remove them from input 'pdata'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 255</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv_subset</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 85</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">rv_subset2</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">subset</span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">"TCGA-02-2485-01"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_n=1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   overlapping columns detected, remove them from input 'pdata'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv_subset2</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 75</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">rv_subset2</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "TCGA-02-2485-01"</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">sum_gene</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getGeneSummary</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">sum_gene</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              gene_id noncircular circular Total</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1: ENSG00000008513           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   2: ENSG00000026036           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3: ENSG00000034677           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4: ENSG00000035687           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5: ENSG00000037897           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ---                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 181: ENSG00000274443           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 182: ENSG00000284188           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 183: ENSG00000285133           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 184: ENSG00000285982           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 185: ENSG00000287908           0        1     1</span>
<span class="r-in"><span class="va">mat_gene</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getGeneSummary</span><span class="op">(</span>return_mat <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in rv$getGeneSummary(return_mat = TRUE):</span> unused argument (return_mat = TRUE)</span>
<span class="r-in"><span class="va">mat_gene</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in eval(expr, envir, enclos):</span> object 'mat_gene' not found</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">sum_cytoband</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getCytobandSummary</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">sum_cytoband</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             band noncircular circular Total</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:  chr12:q13.3           0       17    17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:  chr12:q14.1           0       14    14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:     chr1:q43           0        5     5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:     chr1:q44           0       10    10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: chr20:q13.33           0       30    30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:     chr4:q12           0        4     4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:   chr7:p11.2           0       13    13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:   chr7:p12.1           0        4     4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:   chr7:p12.2           0        5     5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  chr8:q21.11           0       11    11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:  chr8:q21.13           0        4     4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:   chr8:q22.1           0        8     8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:   chr8:q22.2           0       16    16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:   chr8:q22.3           0       19    19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  chr8:q24.13           0        5     5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:  chr8:q24.21           0        6     6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:  chr8:q24.22           0       13    13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:  chr8:q24.23           0        1     1</span>
<span class="r-in"><span class="va">mat_cytoband</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getCytobandSummary</span><span class="op">(</span>return_mat <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in rv$getCytobandSummary(return_mat = TRUE):</span> unused argument (return_mat = TRUE)</span>
<span class="r-in"><span class="va">mat_cytoband</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in eval(expr, envir, enclos):</span> object 'mat_cytoband' not found</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

      </footer></div>

  


  

  </body></html>

