<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. -->
<meta name="color-scheme" content="dark light">
<script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/3/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script>

<title>GCAP workflow for gene-level amplicon prediction — gcap.workflow • gcap</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- Flatly Theme - Light  -->
<link id="css-light" rel="stylesheet" href="https://bootswatch.com/3/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">

<!-- Darkly Theme - Dark -->
<link id="css-dark" rel="stylesheet" href="https://bootswatch.com/3/darkly/bootstrap.css" media="(prefers-color-scheme: dark)">

<!-- preferably CSS -->
<link rel="stylesheet" href="../preferably.css">
<link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)">

  <script src="../darkswitch.js"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="GCAP workflow for gene-level amplicon prediction — gcap.workflow" />
<meta property="og:description" content="GCAP workflow for gene-level amplicon prediction" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>




<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gcap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
        <li>
          <a href="#" id="css-toggle-btn">
            <span class="fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        <li>
  <a href="https://github.com/ShixiangWang/gcap/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
        
        


      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GCAP workflow for gene-level amplicon prediction</h1>
    <small class="dont-index">Source: <a href='https://github.com/ShixiangWang/gcap/blob/master/R/workflow.R'><code>R/workflow.R</code></a></small>
    <div class="hidden name"><code>gcap.workflow.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>GCAP workflow for gene-level amplicon prediction</p>
    </div>

    <pre class="usage"><span class='fu'>gcap.workflow</span><span class='op'>(</span>
  <span class='va'>tumourseqfile</span>,
  <span class='va'>normalseqfile</span>,
  <span class='va'>tumourname</span>,
  <span class='va'>normalname</span>,
  jobname <span class='op'>=</span> <span class='va'>tumourname</span>,
  extra_info <span class='op'>=</span> <span class='cn'>NULL</span>,
  include_type <span class='op'>=</span> <span class='cn'>FALSE</span>,
  genome_build <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"hg38"</span>, <span class='st'>"hg19"</span><span class='op'>)</span>,
  model <span class='op'>=</span> <span class='st'>"XGB32"</span>,
  target <span class='op'>=</span> <span class='st'>"circle"</span>,
  outdir <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/getwd.html'>getwd</a></span><span class='op'>(</span><span class='op'>)</span>,
  result_file_prefix <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"gcap_"</span>, <span class='fu'>uuid</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/uuid/man/UUIDgenerate.html'>UUIDgenerate</a></span><span class='op'>(</span><span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span>,
  allelecounter_exe <span class='op'>=</span> <span class='st'>"~/miniconda3/envs/cancerit/bin/alleleCounter"</span>,
  g1000allelesprefix <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='st'>"~/data/snp/1000G_loci_hg38"</span>,
    <span class='st'>"1kg.phase3.v5a_GRCh38nounref_allele_index_chr"</span><span class='op'>)</span>,
  g1000lociprefix <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='st'>"~/data/snp/1000G_loci_hg38"</span>,
    <span class='st'>"1kg.phase3.v5a_GRCh38nounref_loci_chrstring_chr"</span><span class='op'>)</span>,
  GCcontentfile <span class='op'>=</span> <span class='st'>"~/data/snp/GC_correction_hg38.txt"</span>,
  replictimingfile <span class='op'>=</span> <span class='st'>"~/data/snp/RT_correction_hg38.txt"</span>,
  nthreads <span class='op'>=</span> <span class='fl'>22</span>,
  minCounts <span class='op'>=</span> <span class='fl'>10</span>,
  BED_file <span class='op'>=</span> <span class='cn'>NA</span>,
  probloci_file <span class='op'>=</span> <span class='cn'>NA</span>,
  chrom_names <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>22</span>,
  min_base_qual <span class='op'>=</span> <span class='fl'>20</span>,
  min_map_qual <span class='op'>=</span> <span class='fl'>35</span>,
  penalty <span class='op'>=</span> <span class='fl'>70</span>,
  skip_finished_ASCAT <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>tumourseqfile</th>
      <td><p>Full path to the tumour BAM file.</p></td>
    </tr>
    <tr>
      <th>normalseqfile</th>
      <td><p>Full path to the normal BAM file.</p></td>
    </tr>
    <tr>
      <th>tumourname</th>
      <td><p>Identifier to be used for tumour output files.</p></td>
    </tr>
    <tr>
      <th>normalname</th>
      <td><p>Identifier to be used for normal output files.</p></td>
    </tr>
    <tr>
      <th>jobname</th>
      <td><p>job name, typically an unique name for a tumor-normal pair.</p></td>
    </tr>
    <tr>
      <th>extra_info</th>
      <td><p>(optional) a (file containing) <code>data.frame</code> with 3 columns 'sample'
(must identical to the setting of parameter <code>jobname</code>),
'age' and 'gender'. For gender, should be 'XX' or 'XY',
also could be <code>0</code> for 'XX' and <code>1</code> for 'XY'.</p></td>
    </tr>
    <tr>
      <th>include_type</th>
      <td><p>if <code>TRUE</code>, a fourth column named 'type'
should be included in <code>extra_info</code>, the supported cancer
type includes <code>c("BLCA", "BRCA", "CESC", "COAD", "DLBC", "ESCA", "GBM", "HNSC", "KICH", "KIRC", "KIRP", "LGG", "LIHC", "LUAD", "LUSC", "OV", "PRAD", "SARC", "SKCM", "STAD", "UCEC", "UVM")</code>. Only useful
when use model 'XGB54'.</p></td>
    </tr>
    <tr>
      <th>genome_build</th>
      <td><p>genome build version, should be
one of 'hg38', 'hg19'.</p></td>
    </tr>
    <tr>
      <th>model</th>
      <td><p>model name ("XGB11", "XGB32", "XGB54") or a custom model
from input. 'toy' can be used for test.</p></td>
    </tr>
    <tr>
      <th>target</th>
      <td><p>'circle' and 'nonLinear' (not supported yet) to select model if predict circle
amplicon or non-linear amplicon. <strong>Can be one or both</strong>.</p></td>
    </tr>
    <tr>
      <th>outdir</th>
      <td><p>result output path.</p></td>
    </tr>
    <tr>
      <th>result_file_prefix</th>
      <td><p>file name prefix (without directory path) for storing
final model prediction file in CSV format.
Default a unique file name is generated by UUID approach.</p></td>
    </tr>
    <tr>
      <th>allelecounter_exe</th>
      <td><p>Path to the allele counter executable.</p></td>
    </tr>
    <tr>
      <th>g1000allelesprefix</th>
      <td><p>Prefix path to the 1000 Genomes alleles reference files.</p></td>
    </tr>
    <tr>
      <th>g1000lociprefix</th>
      <td><p>Prefix path to the 1000 Genomes SNP reference files.</p></td>
    </tr>
    <tr>
      <th>GCcontentfile</th>
      <td><p>File containing the GC content around every SNP for increasing window sizes</p></td>
    </tr>
    <tr>
      <th>replictimingfile</th>
      <td><p>File containing replication timing at every SNP for various cell lines (optional)</p></td>
    </tr>
    <tr>
      <th>nthreads</th>
      <td><p>The number of parallel processes for getting allele counts (optional, default=1).</p></td>
    </tr>
    <tr>
      <th>minCounts</th>
      <td><p>Minimum depth required in the normal for a SNP to be considered (optional, default=10).</p></td>
    </tr>
    <tr>
      <th>BED_file</th>
      <td><p>A BED file for only looking at SNPs within specific intervals (optional, default=NA).</p></td>
    </tr>
    <tr>
      <th>probloci_file</th>
      <td><p>A file (chromosome &lt;tab&gt; position; no header) containing specific loci to ignore (optional, default=NA).</p></td>
    </tr>
    <tr>
      <th>chrom_names</th>
      <td><p>A vector containing the names of chromosomes to be
considered (optional, default=1:22).</p></td>
    </tr>
    <tr>
      <th>min_base_qual</th>
      <td><p>Minimum base quality required for a read to be counted (optional, default=20).</p></td>
    </tr>
    <tr>
      <th>min_map_qual</th>
      <td><p>Minimum mapping quality required for a read to be counted (optional, default=35).</p></td>
    </tr>
    <tr>
      <th>penalty</th>
      <td><p>penalty of introducing an additional ASPCF breakpoint (expert parameter, don't adapt unless you know what you're doing)</p></td>
    </tr>
    <tr>
      <th>skip_finished_ASCAT</th>
      <td><p>if <code>TRUE</code>, skipped finished ASCAT calls
to save time.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a list of invisible <code>data.table</code> and corresponding files saved to local machine.</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://orcid.org/0000-0001-9855-7357'>Shixiang Wang</a>.</p>
</div>

<div class="pkgdown">
  <p>Made with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1, using <a href="https://preferably.amirmasoudabdol.name/?source=footer">preferably</a> template.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


