<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>GCAP workflow for gene-level amplicon prediction from ASCN input — gcap.ASCNworkflow • gcap</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="GCAP workflow for gene-level amplicon prediction from ASCN input — gcap.ASCNworkflow"><meta property="og:description" content="Unlike gcap.workflow, this function directly uses the allele-specific
copy number data along with some extra sample information to infer
ecDNA genes."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gcap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ShixiangWang/gcap/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GCAP workflow for gene-level amplicon prediction from ASCN input</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/gcap/blob/HEAD/R/ASCNworkflow.R" class="external-link"><code>R/ASCNworkflow.R</code></a></small>
    <div class="hidden name"><code>gcap.ASCNworkflow.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Unlike <a href="gcap.workflow.html">gcap.workflow</a>, this function directly uses the allele-specific
copy number data along with some extra sample information to infer
ecDNA genes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gcap.ASCNworkflow</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hg38"</span>, <span class="st">"hg19"</span><span class="op">)</span>,</span>
<span>  model <span class="op">=</span> <span class="st">"XGB11"</span>,</span>
<span>  tightness <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  gap_cn <span class="op">=</span> <span class="fl">3L</span>,</span>
<span>  overlap <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  only_oncogenes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  result_file_prefix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gcap_"</span>, <span class="fu">uuid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/uuid/man/UUIDgenerate.html" class="external-link">UUIDgenerate</a></span><span class="op">(</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data</dt>
<dd><p>a <code>data.frame</code> with following columns. The key columns can be obtained
from common allele specific CNV calling software, e.g., ASCAT, Sequenza, FACETS.</p><ul><li><p>chromosome: chromosome names starts with 'chr'.</p></li>
<li><p>start: start position of the segment.</p></li>
<li><p>end: end position of the segment.</p></li>
<li><p>total_cn: total integer copy number of the segment.</p></li>
<li><p>minor_cn: minor allele integer copy number of the segment. Set it
to <code>NA</code> if you don't have this data.</p></li>
<li><p>sample: sample identifier.</p></li>
<li><p>purity: tumor purity of the sample. Set to <code>1</code> if you don't know.</p></li>
<li><p>ploidy (optinal): ploidy value of the sample tumor genome.</p></li>
<li><p>age (optional): age of the case, use along with <code>gender</code>.</p></li>
<li><p>gender (optional): gender of the case, use along with <code>age</code>.</p></li>
<li><p>type (optional): cancer type of the case, use along with <code>age</code> and <code>gender</code>.
Please refer to <a href="gcap.collapse2Genes.html">gcap.collapse2Genes</a> to see the supported cancer types.
This info is only used in 'XGB56' model. If you don't use this model, you
don't need to set it.</p></li>
</ul></dd>


<dt>genome_build</dt>
<dd><p>"hg38" or "hg19".</p></dd>


<dt>model</dt>
<dd><p>model name ("XGB11", "XGB32", "XGB56") or a custom model
from input. 'toy' can be used for test.</p></dd>


<dt>tightness</dt>
<dd><p>a coefficient to times to TCGA somatic CN to set a more strict threshold
as a circular amplicon.
If the value is larger, it is more likely a fCNA assigned to <code>noncircular</code>
instead of <code>circular</code>. <strong>When it is <code>NA</code>, we don't use TCGA somatic CN data as reference</strong>.</p></dd>


<dt>gap_cn</dt>
<dd><p>a gap copy number value.
A gene with copy number above background (<code>ploidy + gap_cn</code> in general) would be treated as focal amplicon.
Smaller, more amplicons.</p></dd>


<dt>overlap</dt>
<dd><p>the overlap percentage on gene.</p></dd>


<dt>only_oncogenes</dt>
<dd><p>if <code>TRUE</code>, only known oncogenes are kept for circular prediction.</p></dd>


<dt>outdir</dt>
<dd><p>result output path.</p></dd>


<dt>result_file_prefix</dt>
<dd><p>file name prefix (without directory path) for storing
final model prediction file in CSV format.
Default a unique file name is generated by UUID approach.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a list of invisible <code>data.table</code> and corresponding files saved to local machine.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ascn"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">ascn</span></span></span>
<span class="r-in"><span><span class="va">rv</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.023972 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.042078 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.045836 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.046465 info [gcap.ASCNworkflow]: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.047074 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.047653 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.048235 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.049627 info [gcap.runASCNBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.051106 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.056154 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.057269]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.058387]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.059062]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.059721]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.068875]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.069699]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.070387]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.071273]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.072292]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.073819]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.076863]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.077574]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.078229]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.078867]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.079757]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.080402]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.081415]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.082148]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.082826]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.089981]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.090739]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.278396]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.279228]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.280159]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.280782]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.281429]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:37.28221]: </span><span style="color: #0000BB; font-weight: bold;">0.225 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.283029 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.285793 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.286969 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.313185 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.31603 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.316838 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.328032 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.347762 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.350537 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.351096 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.351602 info [gcap.runASCNBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.352096 info [gcap.runASCNBuildflow]: None exist, set to NULL.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.352583 info [gcap.runASCNBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.354089 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.354608 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.355794 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.356738 info [collapse_to_genes]: reading reference file /private/var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T/RtmprgZCwj/temp_libpath982677043d72/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.367572 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.381609 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.383266 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.384992 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.393587 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.421358 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.422045 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.422551 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.423037 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The gcap model was developed with xgboost &lt;1.6, it is recommended to install the previous version with command:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> install.packages("https://cran.r-project.org/src/contrib/Archive/xgboost/xgboost_1.5.2.1.tar.gz", repos = NULL, type = "source")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.424724 info [gcap.runPrediction]: using model file XGB_NF11.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.426885 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.59439 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:37] WARNING: src/learner.cc:1203: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:37] WARNING: src/learner.cc:888: Found JSON model saved before XGBoost 1.6, please save the model using current version again. The support for old JSON model will be discontinued in XGBoost 2.3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:37] WARNING: src/learner.cc:553: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:37] WARNING: src/c_api/c_api.cc:935: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.606762 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.607498 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.608005 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.609408 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.60994 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.610434 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:37.61306 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.009441 info [gcap.runScoring]: only keep genes labeled as amplicons in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_prob=0.6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">213</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) <span style="color: #00BB00;">56</span> noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) <span style="color: #00BB00;">157</span> circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.016742 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.208214 info [gcap.ASCNworkflow]: Saving raw prediction result to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_8165f556-1297-11ef-9d84-defcc7b90d10_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.473686 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_8165f556-1297-11ef-9d84-defcc7b90d10_fCNA_records.csv, /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_8165f556-1297-11ef-9d84-defcc7b90d10_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.475865 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.476435 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.476949 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span><span class="va">data</span><span class="op">$</span><span class="va">purity</span> <span class="op">&lt;-</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="va">rv2</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.478852 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.47939 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.479903 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.480395 info [gcap.ASCNworkflow]: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.480883 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.481377 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.481863 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.482919 info [gcap.runASCNBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.483961 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.485276 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.48571]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.486557]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.487186]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.487802]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.489742]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.4904]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.491024]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.491677]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.492344]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.493574]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.495291]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.495949]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.496569]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.497167]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.497905]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.498533]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.499371]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.500043]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.500645]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.506573]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.507328]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.511956]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.51266]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.513405]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.514031]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.514631]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:39.51525]: </span><span style="color: #0000BB; font-weight: bold;">0.03 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.516036 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.518564 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.519737 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.528057 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.530746 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.53153 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.539507 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.556672 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.559449 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.560004 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.560513 info [gcap.runASCNBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.561019 info [gcap.runASCNBuildflow]: None exist, set to NULL.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.561507 info [gcap.runASCNBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.56266 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.563174 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.564273 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.565237 info [collapse_to_genes]: reading reference file /private/var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T/RtmprgZCwj/temp_libpath982677043d72/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.57436 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.587848 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.58932 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.590843 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.59504 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.6199 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.620634 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.621162 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.621693 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The gcap model was developed with xgboost &lt;1.6, it is recommended to install the previous version with command:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> install.packages("https://cran.r-project.org/src/contrib/Archive/xgboost/xgboost_1.5.2.1.tar.gz", repos = NULL, type = "source")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.623603 info [gcap.runPrediction]: using model file XGB_NF11.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.625477 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.628641 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:39] WARNING: src/learner.cc:1203: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:39] WARNING: src/learner.cc:888: Found JSON model saved before XGBoost 1.6, please save the model using current version again. The support for old JSON model will be discontinued in XGBoost 2.3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:39] WARNING: src/learner.cc:553: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:39] WARNING: src/c_api/c_api.cc:935: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.639123 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.639816 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.640344 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.641502 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.642054 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.642602 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:39.644941 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:40.85551 info [gcap.runScoring]: only keep genes labeled as amplicons in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_prob=0.6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">213</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) <span style="color: #00BB00;">56</span> noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) <span style="color: #00BB00;">157</span> circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:40.859995 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.022734 info [gcap.ASCNworkflow]: Saving raw prediction result to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_827ad646-1297-11ef-9d84-defcc7b90d10_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.287503 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_827ad646-1297-11ef-9d84-defcc7b90d10_fCNA_records.csv, /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_827ad646-1297-11ef-9d84-defcc7b90d10_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.289565 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.290098 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.290595 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span><span class="va">data</span><span class="op">$</span><span class="va">age</span> <span class="op">&lt;-</span> <span class="fl">60</span></span></span>
<span class="r-in"><span><span class="va">data</span><span class="op">$</span><span class="va">gender</span> <span class="op">&lt;-</span> <span class="st">"XY"</span></span></span>
<span class="r-in"><span><span class="va">rv3</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB32"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.29273 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.293284 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.293798 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.294298 info [gcap.ASCNworkflow]: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.294793 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.295292 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.295777 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.296847 info [gcap.runASCNBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.297908 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.299265 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.299715]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.300561]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.301188]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.301792]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.303715]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.304373]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.304998]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.305653]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.306308]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.307541]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.309286]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.309945]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.310552]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.311145]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.311878]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.31248]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.313303]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.313975]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.314575]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.320491]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.321192]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.325802]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.326504]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.327257]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.327868]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.328474]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:41.329095]: </span><span style="color: #0000BB; font-weight: bold;">0.029 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.329871 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.332382 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.333563 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.341868 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.344597 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.345387 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.353333 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.370491 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.373326 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.373892 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.374418 info [gcap.runASCNBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.375234 info [gcap.runASCNBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.376508 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.377199 info [gcap.collapse2Genes]: checking data type for age column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.377717 info [gcap.collapse2Genes]: checking data type for gender column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.378224 warn [gcap.collapse2Genes]: non numeric type found, transforming XY to 1, otherwise 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.378934 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.380261 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.381377 info [collapse_to_genes]: reading reference file /private/var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T/RtmprgZCwj/temp_libpath982677043d72/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.39049 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.403852 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.405391 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.406975 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.410956 info [gcap.collapse2Genes]: merging data and extra info</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.414263 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.440598 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.441298 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.441812 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.442308 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The gcap model was developed with xgboost &lt;1.6, it is recommended to install the previous version with command:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> install.packages("https://cran.r-project.org/src/contrib/Archive/xgboost/xgboost_1.5.2.1.tar.gz", repos = NULL, type = "source")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.443985 info [gcap.runPrediction]: using model file XGB_NF32.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.446216 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.450026 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:41] WARNING: src/learner.cc:1203: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:41] WARNING: src/learner.cc:888: Found JSON model saved before XGBoost 1.6, please save the model using current version again. The support for old JSON model will be discontinued in XGBoost 2.3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:41] WARNING: src/learner.cc:553: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:41] WARNING: src/c_api/c_api.cc:935: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.468238 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.468971 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.469564 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.47075 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.471291 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.471819 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:41.47957 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:42.66899 info [gcap.runScoring]: only keep genes labeled as amplicons in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_prob=0.6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">213</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) <span style="color: #00BB00;">56</span> noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) <span style="color: #00BB00;">157</span> circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:42.673483 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:42.853883 info [gcap.ASCNworkflow]: Saving raw prediction result to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_83923f60-1297-11ef-9d84-defcc7b90d10_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.116656 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_83923f60-1297-11ef-9d84-defcc7b90d10_fCNA_records.csv, /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_83923f60-1297-11ef-9d84-defcc7b90d10_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.118027 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.118555 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.119055 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span><span class="co"># If you want to use 'XGB56', you should include 'type' column</span></span></span>
<span class="r-in"><span><span class="va">data</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">"LUAD"</span></span></span>
<span class="r-in"><span><span class="va">rv4</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB56"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.12092 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.12146 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.121971 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.122471 info [gcap.ASCNworkflow]: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.122964 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.123451 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.123933 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.124963 info [gcap.runASCNBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.126007 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.127407 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.127841]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.128682]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.129311]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.129917]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.131849]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.132507]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.133135]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.13379]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.134441]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.13566]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.137358]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.138021]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.138633]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.139234]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.139961]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.140564]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.141392]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.142055]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.142656]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.148563]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.149264]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.153861]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.154545]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.155303]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.155917]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.156522]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:43.157139]: </span><span style="color: #0000BB; font-weight: bold;">0.029 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.157916 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.160435 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.161605 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.169951 info [gcap.extractFeatures]: getting pLOH score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.172626 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.173437 info [gcap.extractFeatures]: generating copy number catalog matrix for fitting signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.181408 info [gcap.extractFeatures]: fitting copy number signature activity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.198551 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.201272 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.201847 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.202369 info [gcap.runASCNBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.203059 info [gcap.runASCNBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.204211 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.204814 info [gcap.collapse2Genes]: one-hot encoding cancer type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.20535 info [gcap.collapse2Genes]: valid cancer types: BLCA,BRCA,CESC,COAD,DLBC,ESCA,GBM,HNSC,KICH,KIRC,KIRP,LGG,LIHC,LUAD,LUSC,OV,PRAD,READ,SARC,SKCM,STAD,THCA,UCEC,UVM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.213348 info [gcap.collapse2Genes]: checking data type for age column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.213972 info [gcap.collapse2Genes]: checking data type for gender column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.214496 warn [gcap.collapse2Genes]: non numeric type found, transforming XY to 1, otherwise 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.215161 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.216336 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.217288 info [collapse_to_genes]: reading reference file /private/var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T/RtmprgZCwj/temp_libpath982677043d72/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.226314 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.239966 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.241439 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.243018 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.247377 info [gcap.collapse2Genes]: merging data and extra info</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.251374 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.278356 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.279099 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.279623 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.280143 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The gcap model was developed with xgboost &lt;1.6, it is recommended to install the previous version with command:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> install.packages("https://cran.r-project.org/src/contrib/Archive/xgboost/xgboost_1.5.2.1.tar.gz", repos = NULL, type = "source")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.281847 info [gcap.runPrediction]: using model file XGB_NF56.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.284115 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.290186 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:43] WARNING: src/learner.cc:1203: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:43] WARNING: src/learner.cc:888: Found JSON model saved before XGBoost 1.6, please save the model using current version again. The support for old JSON model will be discontinued in XGBoost 2.3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:43] WARNING: src/learner.cc:553: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:43] WARNING: src/c_api/c_api.cc:935: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.31985 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.320593 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.321125 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.322299 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.322844 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.323377 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:43.326762 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.340167 info [gcap.runScoring]: only keep genes labeled as amplicons in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_prob=0.6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">213</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) <span style="color: #00BB00;">56</span> noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) <span style="color: #00BB00;">157</span> circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.344667 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.541826 info [gcap.ASCNworkflow]: Saving raw prediction result to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_8493cbb8-1297-11ef-9d84-defcc7b90d10_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.84234 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_8493cbb8-1297-11ef-9d84-defcc7b90d10_fCNA_records.csv, /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_8493cbb8-1297-11ef-9d84-defcc7b90d10_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.843773 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.844326 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.844831 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span><span class="co"># If you only have total integer copy number</span></span></span>
<span class="r-in"><span><span class="va">data</span><span class="op">$</span><span class="va">minor_cn</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span></span>
<span class="r-in"><span><span class="va">rv5</span> <span class="op">&lt;-</span> <span class="fu">gcap.ASCNworkflow</span><span class="op">(</span><span class="va">data</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, model <span class="op">=</span> <span class="st">"XGB11"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.846727 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.847267 info [gcap.ASCNworkflow]:    GCAP ASCN WORKFLOW    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.847779 info [gcap.ASCNworkflow]: =========================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.848275 info [gcap.ASCNworkflow]: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.848758 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.84924 info [gcap.ASCNworkflow]: Step 1: Extract features and collapse features to gene level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.849721 info [gcap.ASCNworkflow]: ============================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.850763 info [gcap.runASCNBuildflow]: extracting sample-level and region-level features</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.851816 info [gcap.extractFeatures]: &gt; Extract features from input ascn_data&lt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.853284 info [gcap.extractFeatures]: generating CopyNumber object in sigminer package</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.853748]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.854591]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg38.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.855223]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.857168]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.857827]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.858452]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.859104]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.859762]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.861017]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.86278]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.863435]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.864041]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 129</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.864636]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.86536]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.865959]: </span><span style="color: #0000BB; font-weight: bold;">Skipped joining adjacent segments with same copy number value.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.866595]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.867186]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.873096]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.8738]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.878293]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.878982]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.879742]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.880354]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.880983]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2024-05-15 16:45:44.8816]: </span><span style="color: #0000BB; font-weight: bold;">0.028 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.88239 info [gcap.extractFeatures]: estimating ploidy from copy number data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.884917 info [gcap.extractFeatures]: checking if input data contains ploidy and if there are NAs should be overwritten</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.886071 info [gcap.extractFeatures]: getting Aneuploidy score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.894402 info [gcap.extractFeatures]: pLOH score is skipped as no LOH data available</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.895098 info [gcap.extractFeatures]: getting CNA burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.895844 info [gcap.extractFeatures]: copy number signature activity quantification is skipped as no LOH data available</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.896357 info [gcap.extractFeatures]: merging data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.897923 info [gcap.extractFeatures]: feature extraction done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.898461 info [gcap.extractFeatures]: now you can modify the result and append 'age' and 'gender' columns to the 'fts_sample' element of result list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.898963 info [gcap.runASCNBuildflow]: checking if input age, gender (and type) columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.899628 info [gcap.runASCNBuildflow]: collapsing all data into gene-level prediction input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.900751 info [gcap.collapse2Genes]: please make sure the first 3 columns of `fts$fts_region` are for chr, start, end.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.901332 info [gcap.collapse2Genes]: one-hot encoding cancer type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.90187 info [gcap.collapse2Genes]: valid cancer types: BLCA,BRCA,CESC,COAD,DLBC,ESCA,GBM,HNSC,KICH,KIRC,KIRP,LGG,LIHC,LUAD,LUSC,OV,PRAD,READ,SARC,SKCM,STAD,THCA,UCEC,UVM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.904945 info [gcap.collapse2Genes]: checking data type for age column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.905482 info [gcap.collapse2Genes]: checking data type for gender column</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.90598 warn [gcap.collapse2Genes]: non numeric type found, transforming XY to 1, otherwise 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.906628 info [gcap.collapse2Genes]: collapsing region-level features to gene-level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.907741 info [collapse_to_genes]: checking input chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.908707 info [collapse_to_genes]: reading reference file /private/var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T/RtmprgZCwj/temp_libpath982677043d72/gcap/extdata/hg38_target_genes.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.917744 info [collapse_to_genes]: finding overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.931422 info [collapse_to_genes]: calculating intersect size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.932857 info [collapse_to_genes]: keeping records with &gt;= 100% overlap ratio with a gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.934381 info [gcap.collapse2Genes]: merging gene-level and sample-level data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.937501 info [gcap.collapse2Genes]: merging data and extra info</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.940237 info [gcap.collapse2Genes]: merging data and prior amplicon frequency data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.964747 info [gcap.collapse2Genes]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.965447 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.965941 info [gcap.ASCNworkflow]: Step 2: Run prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.966414 info [gcap.ASCNworkflow]: =======================</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The gcap model was developed with xgboost &lt;1.6, it is recommended to install the previous version with command:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> install.packages("https://cran.r-project.org/src/contrib/Archive/xgboost/xgboost_1.5.2.1.tar.gz", repos = NULL, type = "source")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.968005 info [gcap.runPrediction]: using model file XGB_NF11.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.969825 info [gcap.runPrediction]: selecting necessary features from input data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.972756 info [gcap.runPrediction]: running prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:44] WARNING: src/learner.cc:1203: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:44] WARNING: src/learner.cc:888: Found JSON model saved before XGBoost 1.6, please save the model using current version again. The support for old JSON model will be discontinued in XGBoost 2.3.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:44] WARNING: src/learner.cc:553: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   If you are loading a serialized model (like pickle in Python, RDS in R) generated by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   older XGBoost, please export the model by calling `Booster.save_model` from that version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   first, then load it back in current version. See:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   for more details about differences between saving model and serializing.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16:45:44] WARNING: src/c_api/c_api.cc:935: `ntree_limit` is deprecated, use `iteration_range` instead.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.984447 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.985204 info [gcap.ASCNworkflow]: Step 3: Run scoring and summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.98578 info [gcap.ASCNworkflow]: ====================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.986978 info [gcap.runScoring]: checking input data type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.987548 info [gcap.runScoring]: checking columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.988071 info [gcap.runScoring]: filtering out records without prob result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:44.990509 info [gcap.runScoring]: joining extra annotation data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:46.125021 info [gcap.runScoring]: only keep genes labeled as amplicons in result fCNA object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_prob=0.6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">213</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) <span style="color: #00BB00;">56</span> noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) <span style="color: #00BB00;">157</span> circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:46.130913 info [gcap.runScoring]: done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:46.323157 info [gcap.ASCNworkflow]: Saving raw prediction result to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_85a39be6-1297-11ef-9d84-defcc7b90d10_prediction_result.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:46.593853 info [gcap.ASCNworkflow]: Saving fCNA records and sample info to /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_85a39be6-1297-11ef-9d84-defcc7b90d10_fCNA_records.csv, /var/folders/gm/lw6z28md2594gcnws_38_9f40000gn/T//RtmpqfHj1B/gcap_85a39be6-1297-11ef-9d84-defcc7b90d10_sample_info.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:46.595834 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:46.596401 info [gcap.ASCNworkflow]:  Done! Thanks for using GCAP ASCN workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;gcap&gt; 2024-05-15 16:45:46.596913 info [gcap.ASCNworkflow]: ===========================================</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># R6 class fCNA --------------------------------</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A &lt;<span style="color: #55FFFF;">fCNA</span>&gt; object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   record: <span style="color: #00BB00;">213</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     case: <span style="color: #00BBBB;">2</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">0</span>) <span style="color: #00BB00;">56</span> noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      |__ (<span style="color: #00BBBB;">2</span>) <span style="color: #00BB00;">157</span> circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ======================</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               sample        band         gene_id total_cn minor_cn ploidy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               &lt;char&gt;      &lt;char&gt;          &lt;char&gt;    &lt;num&gt;    &lt;num&gt;  &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1: TCGA-02-2485-01 chr12:q13.3 ENSG00000111087       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   2: TCGA-02-2485-01 chr12:q13.3 ENSG00000123329       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3: TCGA-02-2485-01 chr12:q13.3 ENSG00000135454       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4: TCGA-02-2485-01 chr12:q13.3 ENSG00000139269       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5: TCGA-02-2485-01 chr12:q13.3 ENSG00000155980       35        2   4.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ---                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 209: TCGA-05-4422-01 chr8:q24.22 ENSG00000155897       15        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 210: TCGA-05-4422-01 chr8:q24.22 ENSG00000155926        7        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 211: TCGA-05-4422-01 chr8:q24.22 ENSG00000165071        7        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 212: TCGA-05-4422-01 chr8:q24.22 ENSG00000184156        7        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 213: TCGA-05-4422-01 chr8:q24.23 ENSG00000131773        7        1   2.09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      background_cn      prob  gene_class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;num&gt;     &lt;num&gt;      &lt;fctr&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1:      4.090000 0.9875240    circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   2:      4.090000 0.9875240    circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3:      4.090000 0.9875240    circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4:      4.090000 0.9875240    circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5:      4.090000 0.9875240    circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ---                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 209:      5.398592 0.9973079    circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 210:      5.355356 0.9183816 noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 211:      5.364956 0.9183816 noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 212:      5.355802 0.9183816 noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 213:      5.339220 0.9183816 noncircular</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">sample_summary</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Key: &lt;sample&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sample purity ploidy AScore       pLOH cna_burden      CN1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;char&gt;  &lt;num&gt;  &lt;num&gt;  &lt;num&gt;      &lt;num&gt;      &lt;num&gt;    &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: TCGA-02-2485-01      1   4.09     13 0.04766306      0.951  0.00000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2: TCGA-05-4422-01      1   2.09     13 0.13041086      0.249 23.84798</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          CN2   CN3   CN4   CN5   CN6      CN7   CN8    CN9  CN10     CN11  CN12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;num&gt; &lt;num&gt; &lt;num&gt; &lt;num&gt; &lt;num&gt;    &lt;num&gt; &lt;num&gt;  &lt;num&gt; &lt;num&gt;    &lt;num&gt; &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: 45.112591     0     0     0     0 9.887409     0  0.000     0 0.000000     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:  2.326143     0     0     0     0 8.939099     0 29.688     0 1.154688     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     CN13  CN14     CN15  CN16  CN17     CN18  CN19    class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    &lt;num&gt; &lt;num&gt;    &lt;num&gt; &lt;num&gt; &lt;num&gt;    &lt;num&gt; &lt;num&gt;   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:     0     0 0.000000     0     0 0.000000     0 circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:     0     0 1.178945     0     0 6.865139     0 circular</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">gene_summary</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">cytoband_summary</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create a subset fCNA</span></span></span>
<span class="r-in"><span><span class="va">rv_subset</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">subset</span><span class="op">(</span><span class="va">total_cn</span> <span class="op">&gt;</span> <span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_prob=0.6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   overlapping columns detected, remove them from input 'pdata'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 213</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv_subset</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 85</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">rv_subset2</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">subset</span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">"TCGA-02-2485-01"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing sample...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   classifying samples with min_prob=0.6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   overlapping columns detected, remove them from input 'pdata'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rv_subset2</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 75</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">rv_subset2</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "TCGA-02-2485-01"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sum_gene</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getGeneSummary</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sum_gene</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              gene_id noncircular circular Total</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               &lt;char&gt;       &lt;int&gt;    &lt;int&gt; &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1: ENSG00000287908           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   2: ENSG00000285982           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   3: ENSG00000285133           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4: ENSG00000284188           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5: ENSG00000274443           0        1     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ---                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 209: ENSG00000066827           1        0     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 210: ENSG00000054796           1        0     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 211: ENSG00000042832           1        0     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 212: ENSG00000042813           1        0     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 213: ENSG00000008513           1        0     1</span>
<span class="r-in"><span><span class="va">mat_gene</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getGeneSummary</span><span class="op">(</span>return_mat <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">mat_gene</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 TCGA.02.2485.01 TCGA.05.4422.01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000287908        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000285982            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000285133        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000284188        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000274443            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000273154            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000273047            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000258830        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000258677            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000258366            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000257921        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000243509            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000240771        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000212993            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000203896            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000203883            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000203667        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000203666        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000198276            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000197769        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000197457            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000197114            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000196700            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000196421            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000196289        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000186106            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000185290        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000183260            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000180938            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000180613        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000180543            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000180287        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000179912        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000179456        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000179397        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000178498        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000177459            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000175606            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000175215        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000175203        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000175197        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000175189        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000174226            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000173728        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000173334            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000171703            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000171700            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000171695            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000169439            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000168672            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000166987        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000166986        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000166908        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000166896        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164961            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164934            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164933            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164932            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164930            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164929            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164924            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164920            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164919            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164776        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164749            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000157404        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156831            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156509            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156486            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156482            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156471            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156469            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156467            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000155980        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000155897            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000155097            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000155096            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000155090            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000155087            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000154589            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000154582            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000153317            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000153310            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000153187        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000147697            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000147669            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000147649            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000147604            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000146731        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000146648        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000143702        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000139269        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000139266        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000139263        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000137558            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000136997            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000135506        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000135454        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000135452        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000135446        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000135439        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000135407        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000134853        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132561            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132554            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132549            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132541            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132294            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000130590            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000130589            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000130584            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000129103        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000125534            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000125531            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000125520            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000125510            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000125508            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000123427        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000123329        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000123297        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000121644        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000121039            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000121005            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000120963            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000117020        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000111087        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000111012        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000109220        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000106153        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104549            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104517            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104490            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104450            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104381            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104375            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104369            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104361            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104356            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104343            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104341            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104324            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000101246            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000101216            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000101213            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000101161            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000101152            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000101150            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000091656            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000083307            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000070756            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000054282        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000048392            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000040341            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000037897        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000035687        circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000034677            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000026036            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000275896     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000270672     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000256222            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000254995            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000249773     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000239789     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000221900     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000215440            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000204983     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000198768            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000188730     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000185811     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000184156            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000178665     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000175895            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000175305            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000170419     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000165071            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164944            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164941            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164938            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000164500     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156172            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156170            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000156162            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000155926            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000154997     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000154978     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132819            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132437     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132436     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132434     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000132297            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000131773            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000129295            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000129292            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124256            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124253            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124237            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124227            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124225            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124222            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124209            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124164            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000124092            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000106078     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000106070     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104419            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104415            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000104413            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000101146            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000066827            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000054796            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000042832            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000042813     noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000008513            &lt;NA&gt;     noncircular</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sum_cytoband</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getCytobandSummary</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sum_cytoband</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             band noncircular circular Total</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;char&gt;       &lt;int&gt;    &lt;int&gt; &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: chr20:q13.33           0       30    30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:   chr8:q22.3           0       19    19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:  chr12:q13.3           0       17    17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:   chr8:q22.2           0       16    16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:  chr12:q14.1           0       14    14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  chr8:q21.11           0       11    11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:     chr1:q44           0       10    10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:   chr8:q22.1           9        8    17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:   chr7:p11.2           7        6    13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  chr8:q24.21           0        6     6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:  chr8:q24.13           0        5     5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:     chr1:q43           0        5     5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:  chr8:q21.13           0        4     4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:     chr4:q12           0        4     4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  chr8:q24.22          11        2    13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: chr20:q13.32           8        0     8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: chr20:q13.31           8        0     8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   chr7:p12.2           5        0     5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:   chr7:p12.1           4        0     4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:     chr7:q34           3        0     3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:  chr8:q24.23           1        0     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             band noncircular circular Total</span>
<span class="r-in"><span><span class="va">mat_cytoband</span> <span class="op">&lt;-</span> <span class="va">rv</span><span class="op">$</span><span class="fu">getCytobandSummary</span><span class="op">(</span>return_mat <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">mat_cytoband</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              TCGA.02.2485.01 TCGA.05.4422.01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr20:q13.33            &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q22.3              &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr12:q13.3         circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q22.2              &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr12:q14.1         circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q21.11             &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr1:q44            circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q22.1              &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr7:p11.2          circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q24.21             &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q24.13             &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr1:q43            circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q21.13             &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr4:q12            circular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q24.22             &lt;NA&gt;        circular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr20:q13.32            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr20:q13.31            &lt;NA&gt;     noncircular</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr7:p12.2       noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr7:p12.1       noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr7:q34         noncircular            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> chr8:q24.23             &lt;NA&gt;     noncircular</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

