<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>GCAP workflow for gene-level amplicon prediction — gcap.workflow • gcap</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="GCAP workflow for gene-level amplicon prediction — gcap.workflow"><meta property="og:description" content="GCAP workflow for gene-level amplicon prediction"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gcap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ShixiangWang/gcap/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GCAP workflow for gene-level amplicon prediction</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/gcap/blob/HEAD/R/workflow.R" class="external-link"><code>R/workflow.R</code></a></small>
    <div class="hidden name"><code>gcap.workflow.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>GCAP workflow for gene-level amplicon prediction</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gcap.workflow</span><span class="op">(</span></span>
<span>  <span class="va">tumourseqfile</span>,</span>
<span>  <span class="va">normalseqfile</span>,</span>
<span>  <span class="va">tumourname</span>,</span>
<span>  <span class="va">normalname</span>,</span>
<span>  jobname <span class="op">=</span> <span class="va">tumourname</span>,</span>
<span>  extra_info <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  include_type <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hg38"</span>, <span class="st">"hg19"</span><span class="op">)</span>,</span>
<span>  model <span class="op">=</span> <span class="st">"XGB11"</span>,</span>
<span>  tightness <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  gap_cn <span class="op">=</span> <span class="fl">3L</span>,</span>
<span>  overlap <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  only_oncogenes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  result_file_prefix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gcap_"</span>, <span class="fu">uuid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/uuid/man/UUIDgenerate.html" class="external-link">UUIDgenerate</a></span><span class="op">(</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  allelecounter_exe <span class="op">=</span> <span class="st">"~/miniconda3/envs/cancerit/bin/alleleCounter"</span>,</span>
<span>  g1000allelesprefix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"~/data/snp/1000G_loci_hg38"</span>,</span>
<span>    <span class="st">"1kg.phase3.v5a_GRCh38nounref_allele_index_chr"</span><span class="op">)</span>,</span>
<span>  g1000lociprefix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"~/data/snp/1000G_loci_hg38"</span>,</span>
<span>    <span class="st">"1kg.phase3.v5a_GRCh38nounref_loci_chrstring_chr"</span><span class="op">)</span>,</span>
<span>  GCcontentfile <span class="op">=</span> <span class="st">"~/data/snp/GC_correction_hg38.txt"</span>,</span>
<span>  replictimingfile <span class="op">=</span> <span class="st">"~/data/snp/RT_correction_hg38.txt"</span>,</span>
<span>  nthreads <span class="op">=</span> <span class="fl">22</span>,</span>
<span>  minCounts <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  BED_file <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  probloci_file <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  chrom_names <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">22</span>,</span>
<span>  min_base_qual <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  min_map_qual <span class="op">=</span> <span class="fl">35</span>,</span>
<span>  penalty <span class="op">=</span> <span class="fl">70</span>,</span>
<span>  skip_finished_ASCAT <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  skip_ascat_call <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>tumourseqfile</dt>
<dd><p>Full path to the tumour BAM file.</p></dd>


<dt>normalseqfile</dt>
<dd><p>Full path to the normal BAM file.</p></dd>


<dt>tumourname</dt>
<dd><p>Identifier to be used for tumour output files.</p></dd>


<dt>normalname</dt>
<dd><p>Identifier to be used for normal output files.</p></dd>


<dt>jobname</dt>
<dd><p>job name, typically an unique name for a tumor-normal pair.</p></dd>


<dt>extra_info</dt>
<dd><p>(optional) a (file containing) <code>data.frame</code> with 3 columns 'sample'
(must identical to the setting of parameter <code>jobname</code>),
'age' and 'gender'. For gender, should be 'XX' or 'XY',
also could be <code>0</code> for 'XX' and <code>1</code> for 'XY'.</p></dd>


<dt>include_type</dt>
<dd><p>if <code>TRUE</code>, a fourth column named 'type'
should be included in <code>extra_info</code>, the supported cancer
type should be described with <a href="https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables/tcga-study-abbreviations" class="external-link">TCGA cancer type abbr.</a>.</p></dd>


<dt>genome_build</dt>
<dd><p>genome build version, should be
one of 'hg38', 'hg19'.</p></dd>


<dt>model</dt>
<dd><p>model name ("XGB11", "XGB32", "XGB56") or a custom model
from input. 'toy' can be used for test.</p></dd>


<dt>tightness</dt>
<dd><p>a coefficient to times to TCGA somatic CN to set a more strict threshold
as a circular amplicon.
If the value is larger, it is more likely a fCNA assigned to <code>noncircular</code>
instead of <code>circular</code>. <strong>When it is <code>NA</code>, we don't use TCGA somatic CN data as reference</strong>.</p></dd>


<dt>gap_cn</dt>
<dd><p>a gap copy number value.
A gene with copy number above background (<code>ploidy + gap_cn</code> in general) would be treated as focal amplicon.
Smaller, more amplicons.</p></dd>


<dt>overlap</dt>
<dd><p>the overlap percentage on gene.</p></dd>


<dt>only_oncogenes</dt>
<dd><p>if <code>TRUE</code>, only known oncogenes are kept for circular prediction.</p></dd>


<dt>outdir</dt>
<dd><p>result output path.</p></dd>


<dt>result_file_prefix</dt>
<dd><p>file name prefix (without directory path) for storing
final model prediction file in CSV format.
Default a unique file name is generated by UUID approach.</p></dd>


<dt>allelecounter_exe</dt>
<dd><p>Path to the allele counter executable.</p></dd>


<dt>g1000allelesprefix</dt>
<dd><p>Prefix path to the 1000 Genomes alleles reference files.</p></dd>


<dt>g1000lociprefix</dt>
<dd><p>Prefix path to the 1000 Genomes SNP reference files.</p></dd>


<dt>GCcontentfile</dt>
<dd><p>File containing the GC content around every SNP for increasing window sizes</p></dd>


<dt>replictimingfile</dt>
<dd><p>File containing replication timing at every SNP for various cell lines (optional)</p></dd>


<dt>nthreads</dt>
<dd><p>The number of parallel processes for getting allele counts (optional, default=1).</p></dd>


<dt>minCounts</dt>
<dd><p>Minimum depth required in the normal for a SNP to be considered (optional, default=10).</p></dd>


<dt>BED_file</dt>
<dd><p>A BED file for only looking at SNPs within specific intervals (optional, default=NA).</p></dd>


<dt>probloci_file</dt>
<dd><p>A file (chromosome &lt;tab&gt; position; no header) containing specific loci to ignore (optional, default=NA).</p></dd>


<dt>chrom_names</dt>
<dd><p>A vector containing the names of chromosomes to be
considered (optional, default=1:22).</p></dd>


<dt>min_base_qual</dt>
<dd><p>Minimum base quality required for a read to be counted (optional, default=20).</p></dd>


<dt>min_map_qual</dt>
<dd><p>Minimum mapping quality required for a read to be counted (optional, default=35).</p></dd>


<dt>penalty</dt>
<dd><p>penalty of introducing an additional ASPCF breakpoint (expert parameter, don't adapt unless you know what you're doing)</p></dd>


<dt>skip_finished_ASCAT</dt>
<dd><p>if <code>TRUE</code>, skipped finished ASCAT calls
to save time.</p></dd>


<dt>skip_ascat_call</dt>
<dd><p>if <code>TRUE</code>, skip calling ASCAT.
This is useful when you have done this step and just want
to run next steps.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a list of invisible <code>data.table</code> and corresponding files saved to local machine.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

