#' GCAP workflow for gene-level amplicon prediction
#'
#' @inheritParams gcap.runASCAT
#' @inheritParams gcap.extractFeatures
#' @inheritParams gcap.collapse2Genes
#' @inheritParams gcap.runPrediction
#' @param feature 'with_type' and 'without_type' to select model whether
#' includes cancer type or not. **Can be one or both**.
#' @param target 'circle' and 'nonLinear' to select model if predict circle
#' amplicon or non-linear amplicon. **Can be one or both**.
#' @param prob_cutoff a value in `[0-1]` to determine the cutoff for labeling
#' response. The default cutoff `0.9` is a balance between F1 score and amplicon
#' load accuracy.
#' @param result_file_prefix file name prefix (without directory path) for storing
#' final model prediction file in CSV format.
#' Default a unique file name is generated by UUID approach.
#' @param extra_info (optional) a (file containing) `data.frame` with 3 columns 'sample',
#' 'age' and 'gender', for including cancer type, check
#' parameter `include_type`. For gender, should be 'XX' or 'XY',
#' also could be `0` for 'XX' and `1` for 'XY'.
#' @return a list of `data.table` and corresponding files saved to local machine.
#' @export
gcap.workflow <- function(tumourseqfile, normalseqfile,
                          tumourname, normalname, jobname = tumourname,
                          extra_info = NULL,
                          include_type = FALSE,
                          genome_build = c("hg38", "hg19"),
                          feature = "without_type",
                          target = c("circle", "nonLinear"),
                          prob_cutoff = 0.9,
                          outdir = getwd(),
                          result_file_prefix = paste0("gcap_", uuid::UUIDgenerate(TRUE)),
                          allelecounter_exe = "~/miniconda3/envs/cancerit/bin/alleleCounter",
                          g1000allelesprefix = file.path(
                            "~/data/snp/1000G_loci_hg38",
                            "1kg.phase3.v5a_GRCh38nounref_allele_index_chr"
                          ),
                          g1000lociprefix = file.path(
                            "~/data/snp/1000G_loci_hg38",
                            "1kg.phase3.v5a_GRCh38nounref_loci_chrstring_chr"
                          ),
                          GCcontentfile = "~/data/snp/GC_correction_hg38.txt",
                          replictimingfile = "~/data/snp/RT_correction_hg38.txt",
                          nthreads = 22,
                          minCounts = 10,
                          BED_file = NA,
                          probloci_file = NA,
                          chrom_names = 1:22,
                          min_base_qual = 20,
                          min_map_qual = 35,
                          penalty = 70,
                          skip_finished_ASCAT = FALSE) {
  genome_build <- match.arg(genome_build)
  # support loopping
  feature <- match.arg(feature, choices = c("with_type", "without_type"), several.ok = TRUE)
  target <- match.arg(target, choices = c("circle", "nonLinear"), several.ok = TRUE)

  lg <- set_logger()
  lg$info("=====================")
  lg$info("   GCAP WORKFLOW")
  lg$info("=====================")
  lg$info()

  lg$info("=====================")
  lg$info("Step 1: Run ASCAT 3.0")
  lg$info("=====================")

  if (is.character(extra_info) && file.exists(extra_info)) {
    extra_info <- data.table::fread(extra_info, header = TRUE)
  }
  totalT <- parallel::detectCores()
  if (nthreads > totalT) {
    lg$info("{nthreads} threads are set but only {totalT} cores are available, reset it to {totalT}")
    nthreads <- totalT
  }

  gcap.runASCAT(
    tumourseqfile,
    normalseqfile,
    tumourname,
    normalname,
    jobname,
    outdir = outdir,
    allelecounter_exe = allelecounter_exe,
    g1000allelesprefix = g1000allelesprefix,
    g1000lociprefix = g1000lociprefix,
    GCcontentfile = GCcontentfile,
    replictimingfile = replictimingfile,
    nthreads = nthreads,
    minCounts = minCounts,
    BED_file = BED_file,
    probloci_file = probloci_file,
    chrom_names = chrom_names,
    gender = if (!is.null(extra_info)) {
      if (is.numeric(extra_info$gender)) {
        ifelse(extra_info$gender == 1, "XY", "XX")
      } else {
        extra_info$gender
      }
    } else {
      "XX"
    },
    min_base_qual = min_base_qual,
    min_map_qual = min_map_qual,
    penalty = penalty,
    skip_finished_ASCAT
  )

  lg$info("checking ASCAT result files")
  ascat_files <- file.path(outdir, paste0(jobname, ".ASCAT.rds"))
  keep_idx <- sapply(ascat_files, function(x) {
    keep <- file.info(x)$size > 2000
    if (!keep) {
      lg$warn("{x} contains a failed ASCAT job, will discard it before next step")
    }
    keep
  })
  ascat_files <- ascat_files[keep_idx]
  if (length(ascat_files) < 1) {
    lg$fatal("no sucessful ASCAT result file to proceed!")
    lg$fatal("check your ASCAT setting before make sure this case could not be used!")
    stop()
  }

  lg$info("========================")
  lg$info("Step 2: Extract features")
  lg$info("========================")

  fts <- gcap.extractFeatures(
    ascat_files,
    genome_build = genome_build
  )

  lg$info("=======================================")
  lg$info("Step 3: Collapse features to gene level")
  lg$info("=======================================")
  model_input <- gcap.collapse2Genes(
    fts, extra_info,
    include_type, genome_build
  )

  lg$info("=======================")
  lg$info("Step 4: Run prediction")
  lg$info("=======================")
  for (f in feature) {
    for (t in target) {
      model_input[[paste0("pred_", t, "_", f)]] <- gcap.runPrediction(
        model_input, f, t
      )
    }
  }
  save_file <- file.path(outdir, paste0(result_file_prefix, "_by_gene.csv"))
  lg$info("Saving result to {save_file}")
  data.table::fwrite(model_input, file = save_file)

  lg$info("=======================")
  lg$info("Step 5: Run scoring")
  lg$info("=======================")
  out <- gcap.runScoring(model_input, prob_cutoff, genome_build)

  save_file <- file.path(outdir, paste0(result_file_prefix, "_by_case.csv"))
  lg$info("Saving result to {save_file}")
  data.table::fwrite(out, file = save_file)

  lg$info("=======================================")
  lg$info(" Done! Thanks for using GCAP workflow")
  lg$info("=======================================")

  invisible(list(
    by_gene = model_input,
    by_case = out
  ))
}
